---
title: "Comparing Mothur to Freshwater DB"
author: "Marian L. Schmidt"
date: "November 4, 2015"
output: html_document
---

```{r knitr_global_options, eval = TRUE, echo = FALSE}
# Set the global options
knitr::opts_chunk$set(echo=FALSE, eval = TRUE, fig.width=6, fig.height=5, fig.align = 'center', warning=FALSE, message=FALSE)
# Set the seed
## runif(1, 0, 10^8) # Run this command once and plug it into the set.seed() function
set.seed(15879966)
```

```{r load libraries and packages}
# Load packages
necessary_packages <-  c("phyloseq", "ggplot2", "ape", "vegan", "plyr", "scales", "grid", "reshape2", "data.table","DESeq2", "dplyr","tidyr", "sciplot","scales", "pgirmess","multcompView", "pander")
# Install the packages that we need
packages <- lapply(necessary_packages, library, character.only = TRUE)

# Make sure to have most updated version of phyloseq
source("https://bioconductor.org/biocLite.R")
biocLite("phyloseq")

# Set the working Directory
setwd("~/Final_PAFL_Trophicstate")

### Source written functions in the file Functions_PAFL.R that is housed within the "Final_PAFL_Trophicstate"
source("Functions_PAFL.R")
```


```{r set colors and theme}
### Set our theme for our plots down the road
set_ggtheme <- theme_set(theme_bw() + 
                           theme(plot.title = element_text(face="bold", size = 12),  #Set the plot title
                                 strip.text.x = element_text(size=10, face="bold"),  #Set the facet titles on x-axis 
                                 strip.text.y = element_text(size=10, face="bold"),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=12),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=12),  #Set the y-axis title
                                 axis.text.x = element_text(colour = "black", size=8),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=8),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="right"))  #Default the legend position to the right

#set phylum plotting colors
phylum.colors <- c(Acidobacteria = "grey26", Actinobacteria = "royalblue", Alphaproteobacteria = "plum2", Armatimonadetes = "red", Bacteroidetes = "darkorange",
                   "BD1-5" = "chartreuse", Betaproteobacteria = "slateblue2", Caldiserica = "black","Candidate_division_BRC1" = "violetred4",
                   "Candidate_division_JS1" = "aquamarine1",
                   "Candidate_division_OD1" = "#6DDE88", "Candidate_division_OP3" = "hotpink", "Candidate_division_OP8" = "goldenrod1", "Candidate_division_OP11" = "chocolate4",
                   "Candidate_division_SR1" = "tan3", "Candidate_division_TM7" = "skyblue1", "Candidate_division_WS3" = "magenta",
                   Chlamydiae="violet", Chlorobi="cyan2", Chloroflexi="darkgreen", Cyanobacteria = "chartreuse3", 
                   Deferribacteres = "slateblue3", "Deinococcus-Thermus" = "violetred", Dictyoglomi = "cornsilk4", Deltaproteobacteria = "deepskyblue", 
                   Elusimicrobia = "yellow3", Epsilonproteobacteria = "lightskyblue", Fibrobacteres = "darkred", Firmicutes = "blue4", FGL7S = "palevioletred1",
                   Fusobacteria = "slateblue1", Gammaproteobacteria = "steelblue4", Gemmatimonadetes="black", GOUTA4 = "plum1", "Hyd24-12" = "sienna2", JTB23 = "seashell2",
                   Lentisphaerae = "yellow1", "NPL-UPA2"="#652926", OC31 = "mediumpurple4", Planctomycetes = "mediumorchid3", Proteobacteria = "deepskyblue",
                   "SHA-109" = "lightsalmon3", SM2F11 = "lightskyblue2", SPOTSOCT00m83 = "orangered",
                   Spirochaetae = "gold3", Tenericutes="pink", Thermotogae = "chocolate1", TA06 = "lightslateblue",TA18 = "rosybrown3", TM6 = "olivedrab",
                   unclassified = "grey", Verrucomicrobia = "purple4", "WCHB1-60" = "palegreen")

##  Do not show scientific notation 
options(scipen = 999, digits = 7)

## Set pander (table) output functions 
panderOptions('table.split.table', Inf)
panderOptions('table.style', 'rmarkdown')
```


```{r data import of Original Mothur data}
### Data import
#sharedfile = "mothur.normalized.shared"
sharedfile <- "~/Final_PAFL_Trophicstate/raw_data/stability.trim.contigs.good.unique.good.filter.precluster.pick.pick.pick.an.unique_list.shared"
taxfile <- "~/Final_PAFL_Trophicstate/raw_data/stability.trim.contigs.good.unique.good.filter.precluster.pick.pick.pick.an.unique_list.0.03.cons.taxonomy"
mothurdata <- import_mothur(mothur_shared_file = sharedfile ,mothur_constaxonomy_file = taxfile )

# Change the taxonomy names
tax_table(mothurdata) <- cbind(tax_table(mothurdata),row.names(tax_table(mothurdata)))
colnames(tax_table(mothurdata)) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species")

# Change some of the sample names to match our metadata sample names
samplenames <- sample_names(mothurdata)
names2 <- gsub("GULL", "GUL", samplenames)
names3 <- gsub("LONG", "LON", names2)
sample_names(mothurdata) <- names3

# Import metadata file and merge with mothurdata object
mapfile <- "~/Final_PAFL_Trophicstate/raw_data/metadata"
map <- import_qiime_sample_data(mapfile)
merge <- merge_phyloseq(mothurdata,map)

# lets look at only samples (removing blanks and mock and samples that didn't amplify)
pruned <- prune_taxa(taxa_sums(merge) > 0, merge)
bact_samples <-subset_taxa(pruned, Kingdom == "Bacteria")
bact_samples <-subset_taxa(bact_samples, Class != "Chloroplast")

###  Time to (1) Sum between replicates and then (2) scale 
# Step 1.  Sum the reads 
sum_reps <- merge_samples(x = bact_samples, group = "dups", fun = "sum") # Sum between replicate samples
sum_reps <- prune_taxa(taxa_sums(sum_reps) > 0, sum_reps) # Remove any OTUs that are 0

# Step 2:  Scale the reads 
sum_scale_matround <- scale_reads(physeq = sum_reps, n = min(sample_sums(sum_reps)), round = "matround")

data_dup_sum_scale_matround <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_sum_scale_matround) <- data_dup_sum_scale_matround$names
data_dup_sum_scale_matround$quadrant <- factor(data_dup_sum_scale_matround$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
# Add a column of Total Sequence sums 
TotalSeqs_sum_reps <- data.frame(rowSums(otu_table(sum_reps)))
TotalSeqs_sum_reps$names <- as.factor(row.names(TotalSeqs_sum_reps))
colnames(TotalSeqs_sum_reps)[1] <- "TotalSeqs_Summed"
data_dup_sum_scale_matround_join<- left_join(data_dup_sum_scale_matround, TotalSeqs_sum_reps, by = "names") 

TotalSeqs_sum_scale_matround <- data.frame(rowSums(otu_table(sum_scale_matround)))
TotalSeqs_sum_scale_matround$names <- as.factor(row.names(TotalSeqs_sum_scale_matround))
colnames(TotalSeqs_sum_scale_matround)[1] <- "TotalSeqs_Scaled"
data_dup_sum_scale_matround_join<- left_join(data_dup_sum_scale_matround_join, TotalSeqs_sum_scale_matround, by = "names") 
row.names(data_dup_sum_scale_matround_join) <- data_dup_sum_scale_matround_join$names

data_dup_sum_scale_matround_join <- sample_data(data_dup_sum_scale_matround_join)  # Metadata in our phyloseq object
sum_scale_matround_otu <- otu_table(sum_scale_matround)  # OTU table in our phyloseq object
sum_scale_matround_tax <- tax_table(sum_scale_matround)  # Taxonomy Table in our phyloseq object
sum_scale_matround <- merge_phyloseq(sum_scale_matround_otu, sum_scale_matround_tax, data_dup_sum_scale_matround_join) 

nowin_silva_scaled <- subset_samples(sum_scale_matround, names != "WINH" & names != "WINH3um")
nowin_silva_scaled <- prune_taxa(taxa_sums(nowin_silva_scaled) > 0, nowin_silva_scaled) # Remove any OTUs that are 0


####  Obtain the top 100 most abundant OTUs
topN <- 50
most_abundant_taxa = sort(taxa_sums(nowin_silva_scaled), TRUE)[1:topN]
print(most_abundant_taxa)
silva_50_OTUs <- prune_taxa(names(most_abundant_taxa), nowin_silva_scaled)

silva_sums <- data.frame(taxa_sums(silva_50_OTUs))

ggplot(silva_sums,aes(x=row.names(silva_sums), y=taxa_sums.silva_50_OTUs.)) + ylab("Number of Sequences per OTU") + scale_x_discrete(expand = c(0,0)) + scale_y_continuous(expand = c(0,0)) + theme_classic() +
  geom_bar(stat="identity",colour="black",fill="skyblue")  + xlab("OTU Rank") + 
  ggtitle("Silva DB: Rank Abundance Curve of Top 50 OTUs") + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

```{r data import of Freshwater DB data}
# Load in the shared file from the mothur FWDB analysis
FWWDB_shared <- "~/Final_PAFL_Trophicstate/Nov6_FWDB_Silva/SoMiLakes_copy_FWDB.files.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.an.unique_list.shared"
# Load in the taxonomy file from the mothur FWDB analysis
FWDB_tax <- "~/Final_PAFL_Trophicstate/Nov6_FWDB_Silva/SoMiLakes_copy_FWDB.files.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.an.unique_list.0.03.cons.taxonomy"
#Create FWDB phyloseq object
FWDB_data <- import_mothur(mothur_shared_file = FWWDB_shared ,mothur_constaxonomy_file = FWDB_tax )

# Change the taxonomy names
#View(tax_table(FWDB_data))
#length(colnames(tax_table(FWDB_data))) # 8 columns 
colnames(tax_table(FWDB_data)) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species", "Strain")
#View(otu_table(FWDB_data))
#sum(row.names(otu_table(FWDB_data)) == row.names(tax_table(FWDB_data))) # OTU names are the same in the tax and otu tables.

#Let's check out the taxonomy table 
tax <- data.frame(tax_table(sub_FWDB))
unique(tax$Class)

#chloroplast <- subset(tax, Class == "Chloroplast")
#unknown_tax <- subset(tax, Kingdom == "unknown")
#eukarya <- subset(tax, Kingdom == "Eukaryota")
#archaea <- subset(tax, Kingdom == "Archaea")
#mitochondria <- subset(tax, Strain == "Mitochondria")

# Remove
sub_FWDB <- subset_taxa(FWDB_data, Kingdom == "Bacteria" &
              Class != "Chloroplast")


#newtax <- sapply(tax, function(x) gsub(pattern = ".__", replacement = "", x)) # apply gsub over all the columns 
#dim(tax); dim(newtax) # Same dimensions

# Change some of the sample names to match our metadata sample names
samplenames_FWDB <- sample_names(sub_FWDB)
names2_FWDB <- gsub("Gull", "GUL", samplenames_FWDB)
names3_FWDB <- gsub("Long", "LON", names2_FWDB)
sample_names(sub_FWDB) <- names3_FWDB

# Import metadata file and merge with sub_FWDB object
mapfile <- "~/Final_PAFL_Trophicstate/raw_data/metadata"
map <- import_qiime_sample_data(mapfile)

#Check the sample names 
names_fw <- sample_names(sub_FWDB)
names_map <- sample_names(map)
length(names_fw); length(names_map)
setdiff(names_fw, names_map)
setdiff(names_map, names_fw)

# Check the sequencing depth of each sample 
sums_FWDB <- data.frame(colSums(otu_table(sub_FWDB)))
colnames(sums_FWDB) <- "Sample_TotalSeqs"
sums_FWDB$sample <- row.names(sums_FWDB)
sums_FWDB <- arrange(sums_FWDB, Sample_TotalSeqs)
ggplot(sums_FWDB, aes(x=reorder(sample, Sample_TotalSeqs), y = Sample_TotalSeqs)) + 
  ylab("Number of Sequences per Sample") +
  geom_bar(stat = "identity", colour="black",fill="cornflowerblue")  + xlab("Sample Name") + ggtitle("Sample Sums") + 
  theme(axis.text.x = element_text(colour = "black", size=8, angle=45, hjust = 1, vjust = 1))  #Set the x-axis labels

pruned_sub_FWDB <- prune_samples(sample_sums(sub_FWDB) > 10000, sub_FWDB)
#names_fw_sub <- sample_names(FWDB_data_sub)
#length(names_fw_sub); length(names_map)
#setdiff(names_fw_sub, names_map)
#setdiff(names_map, names_fw_sub)

# Combine FWDB and mapfiles 
FWDB_no_scale <- merge_phyloseq(pruned_sub_FWDB,map)


# lets look at only samples (removing blanks and mock and samples that didn't amplify)
FW_pruned <- prune_taxa(taxa_sums(FWDB_no_scale) > 0, FWDB_no_scale)
FW_samples <-subset_taxa(FW_pruned, Kingdom == "Bacteria")
FW_samples <-subset_taxa(FW_samples, Class != "Chloroplast")


###  Time to (1) Sum between replicates and then (2) scale 
# Step 1.  Sum the reads 
FW_sum_reps <- merge_samples(x = FW_samples, group = "dups", fun = "sum") # Sum between replicate samples
FW_sum_reps <- prune_taxa(taxa_sums(FW_sum_reps) > 0, FW_sum_reps) # Remove any OTUs that are 0

# Step 2:  Scale the reads 
FWDB_sum_scale_matround <- scale_reads(physeq = FW_sum_reps, n = min(sample_sums(FW_sum_reps)), round = "matround")

data_dup_FWDB_sum_scale_matround <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_FWDB_sum_scale_matround) <- data_dup_FWDB_sum_scale_matround$names
data_dup_FWDB_sum_scale_matround$quadrant <- factor(data_dup_FWDB_sum_scale_matround$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))

# Add a column of Total Sequence sums to data_dup
TotalSeqs_FWDB_sum_reps <- data.frame(rowSums(otu_table(FW_sum_reps)))
TotalSeqs_FWDB_sum_reps$names <- as.factor(row.names(TotalSeqs_FWDB_sum_reps))
colnames(TotalSeqs_FWDB_sum_reps)[1] <- "TotalSeqs_Summed"
data_dup_FWDB_sum_scale_matround_join<- left_join(data_dup_FWDB_sum_scale_matround, TotalSeqs_FWDB_sum_reps, by = "names") 

TotalSeqs_FWDB_sum_scale_matround <- data.frame(rowSums(otu_table(FWDB_sum_scale_matround)))
TotalSeqs_FWDB_sum_scale_matround$names <- as.factor(row.names(TotalSeqs_FWDB_sum_scale_matround))
colnames(TotalSeqs_FWDB_sum_scale_matround)[1] <- "TotalSeqs_Scaled"
data_dup_FWDB_sum_scale_matround_join<- left_join(data_dup_sum_scale_matround, TotalSeqs_FWDB_sum_scale_matround, by = "names") 
row.names(data_dup_FWDB_sum_scale_matround_join) <- data_dup_FWDB_sum_scale_matround_join$names

data_dup_FWDB_sum_scale_matround_join <- sample_data(data_dup_FWDB_sum_scale_matround_join)  # Metadata in our phyloseq object
sum_scale_matround_otu <- otu_table(FWDB_sum_scale_matround)  # OTU table in our phyloseq object
sum_scale_matround_tax <- tax_table(FWDB_sum_scale_matround)  # Taxonomy Table in our phyloseq object
FWDB_sum_scale_matround <- merge_phyloseq(sum_scale_matround_otu, sum_scale_matround_tax, data_dup_FWDB_sum_scale_matround_join) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.


ggplot(data.frame(sum=sample_sums(matround_mean_merge)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="orangered3", binwidth = 500)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean and 'True' Rounding Samples Without Scaling the Reads"), atop("Binwidth = 500 Sequences", ""))))
```


```{r Create the Scaled phyloseq object}
##  Our phyloseq object!
nowin_FWDB_scaled <- subset_samples(FWDB_sum_scale_matround, names != "WINH" & names != "WINH3um")
nowin_FWDB_scaled <- prune_taxa(taxa_sums(nowin_FWDB_scaled) > 0, nowin_FWDB_scaled) # Remove any OTUs that are 0

ggplot(data.frame(sum=sample_sums(nowin_FWDB_scaled)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="thistle1", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Sum & Scale (by 'True' rounding) the Reads"), atop("Binwidth = 20"), ""))) 


########## ADD THE PROTEOBACTERIA TO THE PHYLA
phy <- data.frame(tax_table(nowin_FWDB_scaled))
Phylum <- as.character(phy$Phylum)
Class <- as.character(phy$Class)

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Proteobacteria"){
    Phylum[i] <- Class[i]
  } 
}

phy$Phylum <- Phylum
t <- tax_table(as.matrix(phy))

tax_table(nowin_FWDB_scaled) <- t

# Our phyloseq object!
#nowin_FWDB_scaled 

```

```{r rank abundance curve}


```

```{r comparisons}
silva <- c("Silva Raw", "Merged Silva", "Silva Scaled")
FWDB <- c("FWDB+Silva Raw", "Merged FWDB+Silva", "FWDB+Silva Scaled")

silva_list <- c("bact_samples", "sum_reps", "sum_scale_matround")

fwdb_list <- c( "FW_samples", "FW_sum_reps", "FWDB_sum_scale_matround")
  
compare_silva <- data.frame(matrix(ncol = 6, nrow = 3))
compare_fwdb <- data.frame(matrix(ncol = 6, nrow = 3))

row.names(compare_silva) <- silva_list
row.names(compare_fwdb) <- fwdb_list

colnames(compare_silva) <- c("MinSeqs", "MeanSeqs", "MedianSeqs", "MaxSeqs", "RangeSeqs", "NumOTUs")
colnames(compare_fwdb) <- c("MinSeqs", "MeanSeqs", "MedianSeqs", "MaxSeqs", "RangeSeqs", "NumOTUs")

physeq_stats <- function(physeq){
  out <- c(min(sample_sums(physeq)), 
                  mean(sample_sums(physeq)), 
                  median(sample_sums(physeq)), 
                  max(sample_sums(physeq)),
                  max(sample_sums(physeq)) -min(sample_sums(physeq)),
                  nrow(otu_table(physeq)))
  print(out)
}

compare_silva[1,] <- physeq_stats(bact_samples)
compare_silva[2,] <- physeq_stats(sum_reps)
compare_silva[2,6] <- ncol(otu_table(sum_reps))
compare_silva[3,] <- physeq_stats(sum_scale_matround)
compare_silva[3,6] <- ncol(otu_table(sum_scale_matround))


compare_fwdb[1,] <- physeq_stats(FW_samples)
compare_fwdb[2,] <- physeq_stats(FW_sum_reps)
compare_fwdb[2,6] <- ncol(otu_table(FW_sum_reps))
compare_fwdb[3,] <- physeq_stats(FWDB_sum_scale_matround)
compare_fwdb[3,6] <- ncol(otu_table(FWDB_sum_scale_matround))

panderOptions("digits", 1) # This will set the values to numbers and then will round to the whole number
set.alignment('center', row.names = 'right') # This will align all cells with "center" alignment and the row.names with "right" alignment
pander(compare_silva, caption="Original data with the Silva Database.")

pander(compare_fwdb, caption="New Freshwater Database combined with the Silva Database")
```



```{r overall phylum abundance}
### Calculate the mean relative abundance based on ProdLevel + Quadrant for each PHYLUM 
nosherwin_FWDB_scaled <- subset_samples(nowin_FWDB_scaled, names != "SHEE" & names != "SHEE3um" & names !="SHEH" & names != "SHEH3um") # 537 unique OTUs to Sherman lake
nosherwin_FWDB_scaled <- prune_taxa(taxa_sums(nosherwin_FWDB_scaled) > 0, nosherwin_FWDB_scaled) # Remove any OTUs that are 0


#################################################################################  OVERALL ABUNDANCE PLOT
###############  NO ISOTHERMAL LAKE (SHERMAN) SAMPLES AND NO WINTERGREEN HYPOLIMNION SAMPLES 
good_phylum_proteo <-tax_glom(nosherwin_FWDB_scaled,taxrank = "Phylum")
goodsamps_phylum <- tax_glom(good_phylum_proteo,taxrank = "Phylum")
goodsamps_phy_melt <- psmelt(goodsamps_phylum)  ##  Melt it into a dataframe 
sub_goodsamps_phy_melt <- subset(goodsamps_phy_melt, select = c("Sample", "ProdLevel", "quadrant", "Phylum","Abundance"))
TOTALS <- ddply(sub_goodsamps_phy_melt, c("Sample"), summarise, ##  Let's get the McMurdie and Holmes Scaled  Total for each sample 
                total   = sum(Abundance))   
sub_phy_melt_totals <- merge(sub_goodsamps_phy_melt, TOTALS, by = "Sample")  ### Merge phylum sums and the total numbers -> so we can calculate the Relative Abundance
sub_phy_melt_totals$RelAbundance <- sub_phy_melt_totals$Abundance/sub_phy_melt_totals$total  ## Calculate the relative abundance
sub_phy_melt_totals$PercentAbund <- sub_phy_melt_totals$RelAbundance * 100  ##  Calculate the Percent Abundance


####  Make a new dataframe with the percent abudance within the entire dataset!
phy_stats <- ddply(sub_phy_melt_totals, c("Phylum"), summarise, 
                   N = length(PercentAbund),
                   PercentPhy = mean(PercentAbund),
                   sd   = sd(PercentAbund),
                   se   = sd / sqrt(N))

abund <- subset(phy_stats,PercentPhy > 0.001)  # Only take the phyla that are more than 0.01% abundant

abund$Phylum <- factor(abund$Phylum,levels = c("Bacteroidetes", "Cyanobacteria", "Verrucomicrobia", "Betaproteobacteria", "Actinobacteria", "Planctomycetes",  "Alphaproteobacteria", "Deltaproteobacteria",
                                               "Gammaproteobacteria", "Chloroflexi", "Lentisphaerae", "Firmicutes", "Chlorobi", "Armatimonadetes", "Acidobacteria", "Spirochaetae", "Candidate_division_OD1",
                                               "NPL-UPA2", "Deinococcus-Thermus", "Candidate_division_OP3", "Epsilonproteobacteria", "TM6", "TA18", "Chlamydiae",   "Fibrobacteres", "Candidate_division_SR1", 
                                               "Candidate_division_BRC1", "BD1-5", "Candidate_division_WS3", "Tenericutes", "Elusimicrobia", "Gemmatimonadetes", "WCHB1-60", "Deferribacteres", "Fusobacteria",  
                                               "Candidate_division_TM7",  "SPOTSOCT00m83", "Candidate_division_OP8", "Thermotogae", "Candidate_division_OP11", "Dictyoglomi", "SM2F11", "unclassified"))   

abund$Phylum = with(abund, factor(Phylum, levels = rev(levels(Phylum)))) 


#Relative abundance plot 
ggplot(abund, aes(y=PercentPhy , x=Phylum))  +
  geom_bar(stat="identity", position=position_dodge(),  fill = "cornflowerblue", colour = "black") +
  theme_bw() + ggtitle("Phyla Above 0.1% in All Samples") +
  xlab("Phylum") + ylab("Mean Relative Abundance (%)") +
  geom_errorbar(aes(ymin = PercentPhy -se, ymax = PercentPhy +se), width = 0.25) + 
  coord_flip() 
```




```{r abundance bar plots}
##################################################################################### ABUNDANCE PLOTS ##################  Working up to FIGURE S3 and FIGURE S4
##################################################################################### ABUNDANCE PLOTS ##################  Working up to FIGURE S3 and FIGURE S4
##################################################################################### ABUNDANCE PLOTS ##################  Working up to FIGURE S3 and FIGURE S4

##  Calculate the Phylum abundance based on the ProdLevel and the filter fraction.  
prod_phylum_FWDB_stats <- ddply(sub_phy_melt_totals, c("ProdLevel","quadrant", "Phylum"), summarise, 
                                N = length(PercentAbund),
                                mean_abundance = mean(PercentAbund),
                                sd   = sd(PercentAbund),
                                se   = sd / sqrt(N))
###  Only the phylum 
abund_only_by_phylum <- ddply(prod_phylum_FWDB_stats, c("Phylum"), summarise, 
                              N = length(mean_abundance),
                              phylum_mean = mean(mean_abundance))

abund_only_by_phylum <-arrange(abund_only_by_phylum, desc(phylum_mean)) # abund_only_by_phylum[with(abund_only_by_phylum, order(-phylum_mean)), ]

phy_order <- as.character(abund_only_by_phylum$Phylum)  # THIS IS A VECTOR OF THE PHYLUM IN THE ORDER THAT WE WANT

### TOP 25 PHYLA!
top15phy <- prod_phylum_FWDB_stats[prod_phylum_FWDB_stats$Phylum %in% phy_order[1:15], ]

phy15_order <- as.character(abund_only_by_phylum$Phylum)[1:15]

# ********  NOTE!!!   Be sure to check that the following factor order is the same order as phy15_order
top15phy$Phylum <- factor(top15phy$Phylum,levels = c("Bacteroidetes", "Cyanobacteria", "Verrucomicrobia", "Betaproteobacteria", "Actinobacteria", "Planctomycetes", "Alphaproteobacteria", "unclassified",          
                                                     "Deltaproteobacteria", "Gammaproteobacteria", "Chloroflexi", "Lentisphaerae", "Armatimonadetes",  "Firmicutes",  "Chlorobi"))
  
top15phy$quadrant <- factor(top15phy$quadrant,levels = c("Free Epilimnion", "Particle Epilimnion", "Free Hypolimnion", "Particle Hypolimnion"))

top15phy$Phylum = with(top15phy, factor(Phylum, levels = rev(levels(Phylum))))

#### ADDING NO SPACES FOR FREE LIVING and ADDING 2 SPACE FOR PARICLE ASSOCIATED AT THE END
top15phy$quadrant <- as.character(top15phy$quadrant)
top15phy$quadrant[top15phy$quadrant=="Particle Epilimnion"] <- "Epilimnion   "
top15phy$quadrant[top15phy$quadrant=="Free Epilimnion"]   <- "Epilimnion"
top15phy$quadrant[top15phy$quadrant=="Particle Hypolimnion"]   <- " Hypolimnion  "
top15phy$quadrant[top15phy$quadrant=="Free Hypolimnion"]   <- "Hypolimnion"

#### ADDING NO SPACES FOR PRODUCTIVE and ADDING 1 SPACE FOR UNPRODUCTIVE
top15phy$prod <- top15phy$ProdLevel
top15phy$prod <- as.character(top15phy$prod)
top15phy$prod[top15phy$prod =="Productive"]   <- ""
top15phy$prod[top15phy$prod =="Unproductive"]   <- " "

for(i in 1:length(top15phy$prod)){
  top15phy$prod_quad [i]<-paste(as.character(top15phy$prod[i]),as.character(top15phy$quadrant[i]))}

top15phy$prod_quad <- factor(top15phy$prod_quad,levels = c(" Epilimnion", " Hypolimnion", " Epilimnion   ", "  Hypolimnion  ", 
                                                           "  Epilimnion", "  Hypolimnion", "  Epilimnion   ",  "   Hypolimnion  "))

phy.colors.whew <- c(Acidobacteria = "grey26", Actinobacteria = "palevioletred2", Alphaproteobacteria = "steelblue4", Armatimonadetes = "red", Bacteroidetes = "darkorange",
                     "BD1-5" = "chartreuse", Betaproteobacteria = "royalblue", Caldiserica = "black","Candidate_division_BRC1" = "red",
                     "Candidate_division_JS1" = "aquamarine1",
                     "Candidate_division_OD1" = "#6DDE88", "Candidate_division_OP3" = "yellow1", "Candidate_division_OP8" = "goldenrod1", "Candidate_division_OP11" = "chocolate4",
                     "Candidate_division_SR1" = "tan3", "Candidate_division_TM7" = "skyblue1", "Candidate_division_WS3" = "magenta",
                     Chlamydiae="violet", Chlorobi="cyan2", Chloroflexi="darkgreen", Cyanobacteria = "chartreuse3", 
                     Deferribacteres = "slateblue3", "Deinococcus-Thermus" = "violetred", Dictyoglomi = "cornsilk4", Deltaproteobacteria = "deepskyblue", 
                     Elusimicrobia = "violetred4", Epsilonproteobacteria = "lightskyblue", Fibrobacteres = "hotpink", Firmicutes = "blue4", FGL7S = "palevioletred1",
                     Fusobacteria = "slateblue1", Gammaproteobacteria = "plum2", Gemmatimonadetes="black", GOUTA4 = "plum1", "Hyd24-12" = "sienna2", JTB23 = "seashell2",
                     Lentisphaerae = "yellow1", "NPL-UPA2"="royalblue", OC31 = "mediumpurple4", Planctomycetes = "mediumorchid3", Proteobacteria = "deepskyblue",
                     "SHA-109" = "lightsalmon3", SM2F11 = "lightskyblue2", SPOTSOCT00m83 = "orangered",
                     Spirochaetae = "gold3", Tenericutes="pink", Thermotogae = "chocolate1", TA06 = "lightslateblue",TA18 = "rosybrown3", TM6 = "olivedrab",
                     unclassified = "grey", Verrucomicrobia = "purple4", "WCHB1-60" = "palegreen")


###########################  Here we will use GTable to manually create the labels. 
library(gtable)
# get gtable object
top15plot <- ggplot(top15phy, aes(y=mean_abundance , x=Phylum, fill=Phylum)) + #coord_cartesian(xlim = c(0, 30)) + 
  guides(fill = guide_legend(reverse=TRUE)) + ggtitle("") + 
  geom_bar(stat="identity", position=position_dodge()) + #theme_classic() +
  geom_bar(stat="identity", position=position_dodge(), colour = "black", show_guide = FALSE) +
  facet_wrap(~ prod_quad, ncol = 8) + xlab("Top 15 Most Abundant Phyla") +
  scale_fill_manual(values = phy.colors.whew,name="Phylum") + 
  geom_errorbar(aes(ymin = mean_abundance -se, ymax = mean_abundance +se), width = 0.25, color = "black") + 
  ylab("Mean Percent Relative Abundance (%)") + coord_flip() + theme_bw() 


z <- ggplot_gtable(ggplot_build(top15plot))

# add label for top strip
z <- gtable_add_rows(z, z$heights[[3]], 2)
z <- gtable_add_grob(z, list(rectGrob(gp = gpar(col = NA, linetype = 1, fill = gray(0.7))),
                             textGrob("Free-Living", gp = gpar(fontsize = 14, col = "black"))),
                     3, 4, 3, 7, name = paste(runif(2)))

z <- gtable_add_grob(z, list(rectGrob(gp = gpar(col = NA, linetype = 1, fill = gray(0.7))),
                             textGrob("Particle-Associated", gp = gpar(fontsize = 14, col = "black"))),
                     3, 9, 3, 13, name = paste(runif(2)))

z <- gtable_add_grob(z, list(rectGrob(gp = gpar(col = NA, linetype = 1, fill = gray(0.6))),
                             textGrob("High-Nutrient", gp = gpar(fontsize = 14, col = "black"))),
                     2, 4, 2, 13, name = paste(runif(2)))

z <- gtable_add_grob(z, list(rectGrob(gp = gpar(col = NA, linetype = 1, fill = gray(0.7))),
                             textGrob("Free-Living", gp = gpar(fontsize = 14, col = "black"))),
                     3, 15, 3, 19, name = paste(runif(2)))

z <- gtable_add_grob(z, list(rectGrob(gp = gpar(col = NA, linetype = 1, fill = gray(0.7))),
                             textGrob("Particle-Associated", gp = gpar(fontsize = 14, col = "black"))),
                     3, 22, 3, 25, name = paste(runif(2)))

z <- gtable_add_grob(z, list(rectGrob(gp = gpar(col = NA, linetype=1, fill = gray(0.6))),
                             textGrob("Low-Nutrient", gp = gpar(fontsize = 14, col = "black"))),
                     2, 15, 2, 25, name = paste(runif(2)))

# add margins
z <- gtable_add_cols(z, unit(1/8, "line"), 7)
z <- gtable_add_rows(z, unit(1/8, "line"), 3)

#####  Plotting FIGURE S3  #####  Plotting FIGURE S3  #####  Plotting FIGURE S3  #####  Plotting FIGURE S3  #####  Plotting FIGURE S3
#tiff(filename="~/Final_PAFL_Trophicstate/Final_Figures/Fig.S3_Abundance_top15_gtable.tiff", width= 40, height=25, units= "cm", pointsize= 10, res=200)
# draw it
grid.newpage()
grid.draw(z)
#dev.off()
```


```{r abundance plots comparing 2}
#  Prep the data frame 
bicompare_phy <- subset(goodsamps_phy_melt, select = c("Sample", "filter", "ProdLevel", "quadrant", "Phylum","Abundance"))

bi_TOTALS <- ddply(bicompare_phy, c("Sample"), summarise, ##  Let's get the McMurdie and Holmes Scaled  Total for each sample 
                total   = sum(Abundance))   
bicompare_totals <- merge(bicompare_phy, bi_TOTALS, by = "Sample")  ### Merge phylum sums and the total numbers -> so we can calculate the Relative Abundance
bicompare_phy$RelAbundance <- bicompare_phy$Abundance/bicompare_totals$total  ## Calculate the relative abundance
bicompare_phy$PercentAbund <- bicompare_phy$RelAbundance * 100  ##  Calculate the Percent Abundance

# Compare particle associated to free living 
#filter_phy <- ddply(bicompare_phy, c("filter", "Phylum"), summarise, 
#                                N = length(PercentAbund),
#                                mean_abundance = mean(PercentAbund),
#                                sd   = sd(PercentAbund),
#                                se   = sd / sqrt(N))


bicompare_phy <- bicompare_phy[bicompare_phy$Phylum %in% phy_order[1:15], ]

# ********  NOTE!!!   Be sure to check that the following factor order is the same order as phy15_order
bicompare_phy$Phylum <- factor(bicompare_phy$Phylum,levels = c("Bacteroidetes", "Cyanobacteria", "Verrucomicrobia", "Betaproteobacteria", "Actinobacteria", "Planctomycetes", "Alphaproteobacteria", "unclassified",          
                                                     "Deltaproteobacteria", "Gammaproteobacteria", "Chloroflexi", "Lentisphaerae", "Armatimonadetes",  "Firmicutes",  "Chlorobi"))



ggplot(bicompare_phy, aes(y=PercentAbund , x=Phylum, fill=filter)) + #coord_cartesian(xlim = c(0, 30)) + 
  ggtitle("") + geom_boxplot() +  ylab("Mean Percent Relative Abundance (%)") + 
  theme_bw() + theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))



ggplot(bicompare_phy, aes(y=PercentAbund , x=Phylum, fill=ProdLevel)) + #coord_cartesian(xlim = c(0, 30)) + 
  ggtitle("") + geom_boxplot() +  ylab("Mean Percent Relative Abundance (%)") + 
  theme_bw() + theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))







