---
title: "Analysis Revisions after 1st Reviews"
author: "Marian L. Schmidt"
date: "October 27, 2015"
output: html_document
---

```{r knitr_global_options, eval = TRUE, echo = FALSE}
# Set the global options
knitr::opts_chunk$set(echo=FALSE, eval = TRUE, fig.width=6, fig.height=5, fig.align = 'center', warning=FALSE, message=FALSE)
# Set the seed
## runif(1, 0, 10^8) # Run this command once and plug it into the set.seed() function
set.seed(15879966)
```

```{r load libraries and packages}
# Load packages
necessary_packages <-  c("phyloseq", "ggplot2", "ape", "vegan", "plyr", "scales", "grid", "reshape2", "data.table","DESeq2", "dplyr","tidyr", "sciplot","scales", "pgirmess","multcompView", "pander")
# Install the packages that we need
packages <- lapply(necessary_packages, library, character.only = TRUE)

# Make sure to have most updated version of phyloseq
source("https://bioconductor.org/biocLite.R")
biocLite("phyloseq")

### Set the Working Directory  
###  Within the "Final_PAFL_Trophicstate" directory there should be the following 2 sub-directories:
        #   1.  "raw_data":  includes 5 files:  taxonomy table, OTU (shared) table, "metadata", "duplicate_metadata.txt", "AllLakes_depthprofile.csv" 
        #   2.  "alpha_data":  includes 2 files:  "InvSimpson100" and "ObservedRichness100"  (These files are made between lines 261-295 and take ~20 minutes to calculate.  These files are added for reviewer convenience.)
setwd("~/Final_PAFL_Trophicstate")

### Source written functions in the file Functions_PAFL.R that is housed within the "Final_PAFL_Trophicstate"
source("Functions_PAFL.R")
```


```{r set colors and theme}
### Set our theme for our plots down the road
set_ggtheme <- theme_set(theme_bw() + 
                           theme(plot.title = element_text(face="bold", size = 12),  #Set the plot title
                                 strip.text.x = element_text(size=10, face="bold"),  #Set the facet titles on x-axis 
                                 strip.text.y = element_text(size=10, face="bold"),  #Set the facet titles on x-axis 
                                 strip.background = element_blank(),  #Set the facet background to no background
                                 axis.title.x = element_text(face="bold", size=12),  #Set the x-axis title
                                 axis.title.y = element_text(face="bold", size=12),  #Set the y-axis title
                                 axis.text.x = element_text(colour = "black", size=8),  #Set the x-axis labels
                                 axis.text.y = element_text(colour = "black", size=8),  #Set the y-axis labels
                                 legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
                                 legend.text = element_text(size = 8),  #Set the legend text
                                 legend.position="right"))  #Default the legend position to the right

#set phylum plotting colors
phylum.colors <- c(Acidobacteria = "grey26", Actinobacteria = "royalblue", Alphaproteobacteria = "plum2", Armatimonadetes = "red", Bacteroidetes = "darkorange",
                   "BD1-5" = "chartreuse", Betaproteobacteria = "slateblue2", Caldiserica = "black","Candidate_division_BRC1" = "violetred4",
                   "Candidate_division_JS1" = "aquamarine1",
                   "Candidate_division_OD1" = "#6DDE88", "Candidate_division_OP3" = "hotpink", "Candidate_division_OP8" = "goldenrod1", "Candidate_division_OP11" = "chocolate4",
                   "Candidate_division_SR1" = "tan3", "Candidate_division_TM7" = "skyblue1", "Candidate_division_WS3" = "magenta",
                   Chlamydiae="violet", Chlorobi="cyan2", Chloroflexi="darkgreen", Cyanobacteria = "chartreuse3", 
                   Deferribacteres = "slateblue3", "Deinococcus-Thermus" = "violetred", Dictyoglomi = "cornsilk4", Deltaproteobacteria = "deepskyblue", 
                   Elusimicrobia = "yellow3", Epsilonproteobacteria = "lightskyblue", Fibrobacteres = "darkred", Firmicutes = "blue4", FGL7S = "palevioletred1",
                   Fusobacteria = "slateblue1", Gammaproteobacteria = "steelblue4", Gemmatimonadetes="black", GOUTA4 = "plum1", "Hyd24-12" = "sienna2", JTB23 = "seashell2",
                   Lentisphaerae = "yellow1", "NPL-UPA2"="#652926", OC31 = "mediumpurple4", Planctomycetes = "mediumorchid3", Proteobacteria = "deepskyblue",
                   "SHA-109" = "lightsalmon3", SM2F11 = "lightskyblue2", SPOTSOCT00m83 = "orangered",
                   Spirochaetae = "gold3", Tenericutes="pink", Thermotogae = "chocolate1", TA06 = "lightslateblue",TA18 = "rosybrown3", TM6 = "olivedrab",
                   unclassified = "grey", Verrucomicrobia = "purple4", "WCHB1-60" = "palegreen")

##  Do not show scientific notation 
options(scipen = 999, digits = 7)

## Set pander (table) output functions 
panderOptions('table.split.table', Inf)
panderOptions('table.style', 'rmarkdown')

```


```{r data import}
### Data import
#sharedfile = "mothur.normalized.shared"
sharedfile <- "~/Final_PAFL_Trophicstate/raw_data/stability.trim.contigs.good.unique.good.filter.precluster.pick.pick.pick.an.unique_list.shared"
taxfile <- "~/Final_PAFL_Trophicstate/raw_data/stability.trim.contigs.good.unique.good.filter.precluster.pick.pick.pick.an.unique_list.0.03.cons.taxonomy"
mothurdata <- import_mothur(mothur_shared_file = sharedfile ,mothur_constaxonomy_file = taxfile )

# Change the taxonomy names
tax_table(mothurdata) <- cbind(tax_table(mothurdata),row.names(tax_table(mothurdata)))
colnames(tax_table(mothurdata)) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species")

# Change some of the sample names to match our metadata sample names
samplenames <- sample_names(mothurdata)
names2 <- gsub("GULL", "GUL", samplenames)
names3 <- gsub("LONG", "LON", names2)
sample_names(mothurdata) <- names3

# Import metadata file and merge with mothurdata object
mapfile <- "~/Final_PAFL_Trophicstate/raw_data/metadata"
map <- import_qiime_sample_data(mapfile)
merge <- merge_phyloseq(mothurdata,map)

# lets look at only samples (removing blanks and mock and samples that didn't amplify)
pruned <- prune_taxa(taxa_sums(merge) > 0, merge)
bact_samples <-subset_taxa(pruned, Kingdom == "Bacteria")
#nrow(otu_table(pruned)) - nrow(otu_table(bact_samples))  ## Check how many rows were not bacteria
bact_samples <-subset_taxa(bact_samples, Class != "Chloroplast")
#nrow(otu_table(bact_samples)) - nrow(otu_table(bact_samples2))  # Check how many rows were chloroplasts

#Summary of object
#bact_samples

## Raw Sample read counts
# Histogram of RAW sample read counts
raw_plot <- ggplot(data.frame(sum=sample_sums(bact_samples)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="dodgerblue2", binwidth = 750)  + xlab("Total Sequences") +
   ggtitle(expression(atop(bold("Raw Samples"), atop("Binwidth = 750 Sequences", "")))); raw_plot
```



```{r Mean Samples}
################################################################
### Take the average for samples that have a replicate
noscale_merge <- merge_samples_mean(bact_samples, group = "dups", round = "none")  # Here we simply take the Mean between replicate samples 
noscale_merge <- prune_taxa(taxa_sums(noscale_merge) > 0, noscale_merge)

#######  The above functions mess up the metadata - let's fix it!
data_dup <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup) <- data_dup$names
data_dup$quadrant <- factor(data_dup$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
data_dup <- sample_data(data_dup)  # Metadata in our phyloseq object
scale_otu <- otu_table(noscale_merge)  # OTU table in our phyloseq object
scale_tax <- tax_table(noscale_merge)  # Taxonomy Table in our phyloseq object
raw_merged <- merge_phyloseq(scale_otu, scale_tax, data_dup) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.

##  Sample read counts with merging samples WITHOUT SCALING!!!! 
# Histogram of RAW sample read counts
mean_plot <- ggplot(data.frame(sum=sample_sums(noscale_merge)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="violetred", binwidth = 500)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean Samples Without Scaling the Reads"), atop("Binwidth = 500 Sequences", ""))))
```

```{r Mean and Rounded Samples}
### Take the average for samples that have a replicate
round_mean_merge <- merge_samples_mean(bact_samples, group = "dups", round = "round")  # Here we calculate the "r" mean between the replicate samples 
round_mean_merge <- prune_taxa(taxa_sums(round_mean_merge) > 0, round_mean_merge)

#######  The above functions mess up the metadata - let's fix it!
data_dup_round <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_round) <- data_dup_round$names
data_dup_round$quadrant <- factor(data_dup_round$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
data_dup_round <- sample_data(data_dup_round)  # Metadata in our phyloseq object
scale_otu <- otu_table(round_mean_merge)  # OTU table in our phyloseq object
scale_tax <- tax_table(round_mean_merge)  # Taxonomy Table in our phyloseq object
round_mean_merge <- merge_phyloseq(scale_otu, scale_tax, data_dup_round) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.

##  Sample read counts with merging samples WITHOUT SCALING!!!! 
# Histogram of RAW sample read counts
mean_round_plot <- ggplot(data.frame(sum=sample_sums(round_mean_merge)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="tomato", binwidth = 500)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean and 'R' rounded Samples Without Scaling the Reads"), atop("Binwidth = 500 Sequences", ""))))
```

```{r Mean and Floored Samples, fig.width = 15, fig.height = 6}
################################################################
### Take the average for samples that have a replicate
floor_mean_merge <- merge_samples_mean(bact_samples, group = "dups", round = "floor")  # Calculate the mean and floor the values
floor_mean_merge <- prune_taxa(taxa_sums(floor_mean_merge) > 0, floor_mean_merge)

#######  The above functions mess up the metadata - let's fix it!
data_dup_floor <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_floor) <- data_dup_floor$names
data_dup_floor$quadrant <- factor(data_dup_floor$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
data_dup_floor <- sample_data(data_dup_floor)  # Metadata in our phyloseq object
scale_otu <- otu_table(floor_mean_merge)  # OTU table in our phyloseq object
scale_tax <- tax_table(floor_mean_merge)  # Taxonomy Table in our phyloseq object
floor_mean_merge <- merge_phyloseq(scale_otu, scale_tax, data_dup_floor) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.

# Histogram of sample read counts
mean_floor_plot <- ggplot(data.frame(sum=sample_sums(floor_mean_merge)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="darkorange", binwidth = 500)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean and Floored Samples Without Scaling the Reads"), atop("Binwidth = 500 Sequences", ""))))
```



```{r Mean and True Rounding of Samples, fig.height = 12, fig.width = 12}
################################################################
### Take the average for samples that have a replicate
matround_mean_merge <- merge_samples_mean(bact_samples, group = "dups", round = "matround")  # Calculate the "true mean"
matround_mean_merge <- prune_taxa(taxa_sums(matround_mean_merge) > 0, matround_mean_merge)

#######  The above functions mess up the metadata - let's fix it!
data_dup_matround <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_matround) <- data_dup_matround$names
data_dup_matround$quadrant <- factor(data_dup_matround$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
data_dup_matround <- sample_data(data_dup_matround)  # Metadata in our phyloseq object
scale_otu <- otu_table(matround_mean_merge)  # OTU table in our phyloseq object
scale_tax <- tax_table(matround_mean_merge)  # Taxonomy Table in our phyloseq object
matround_mean_merge <- merge_phyloseq(scale_otu, scale_tax, data_dup_matround) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.

# Histogram of RAW sample read counts
mean_matround_plot <- ggplot(data.frame(sum=sample_sums(matround_mean_merge)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="orangered3", binwidth = 500)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean and 'True' Rounding Samples Without Scaling the Reads"), atop("Binwidth = 500 Sequences", ""))))

multiplot(mean_plot, mean_round_plot, mean_floor_plot, mean_matround_plot, cols = 2)
```



## Influence of Merging Samples and Scaling the Reads 
```{r scale reads by rounding and flooring}
### Now let's scale our read counts from our MERGED RAW data
scaled_merged_round <- scale_reads(physeq = raw_merged, n = min(sample_sums(raw_merged)), round = "round")

# Histogram of sample read counts
mean_scaled_round_plot <- ggplot(data.frame(sum=sample_sums(scaled_merged_round)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="plum2", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean Samples with Scaling (and 'r' rounding) the Reads"), atop("Binwidth = 20"), ""))) 

# Scaled with a "true" rounding function 
scaled_merged_matround <- scale_reads(physeq = raw_merged, n = min(sample_sums(raw_merged)), round = "matround")
# Histogram of sample read counts
mean_scaled_matround_plot <- ggplot(data.frame(sum=sample_sums(scaled_merged_matround)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="plum4", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean Samples with Scaling (and 'true' rounding) the Reads"), atop("Binwidth = 20"), ""))) 


## Now let's scale our read counts from our MERGED RAW data
scaled_merged_floor <- scale_reads(physeq = raw_merged, n = min(sample_sums(raw_merged)), round = "floor")

# Histogram of sample read counts 
mean_scaled_floor_plot <- ggplot(data.frame(sum=sample_sums(scaled_merged_floor)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="purple", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Mean Samples with Scaling (and flooring) the Reads"), atop("Binwidth = 20"), ""))) 
```



```{r sum-scale }
### Now let's 1. sum between replicates.  

# Step 1.  Sum the reads 
sum_reps <- merge_samples(x = bact_samples, group = "dups", fun = "sum") # Sum between replicate samples
sum_reps <- prune_taxa(taxa_sums(sum_reps) > 0, sum_reps) # Remove any OTUs that are 0

# The above function ruins the sample data :/  
data_dup_sum_reps <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_sum_reps) <- data_dup_sum_reps$names
data_dup_sum_reps$quadrant <- factor(data_dup_sum_reps$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
# Add a column of Total Sequence sums 
TotalSeqs_sum_reps <- data.frame(rowSums(otu_table(sum_reps)))
TotalSeqs_sum_reps$names <- as.factor(row.names(TotalSeqs_sum_reps))
colnames(TotalSeqs_sum_reps)[1] <- "TotalSeqs_Summed"
data_dup_sum_reps<- left_join(data_dup_sum_reps, TotalSeqs_sum_reps, by = "names")
row.names(data_dup_sum_reps) <- data_dup_sum_reps$names
data_dup_sum_reps <- sample_data(data_dup_sum_reps)  # Metadata in our phyloseq object
scale_otu <- otu_table(sum_reps)  # OTU table in our phyloseq object
scale_tax <- tax_table(sum_reps)  # Taxonomy Table in our phyloseq object
sum_reps <- merge_phyloseq(scale_otu, scale_tax, data_dup_sum_reps) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.

# Step 3:  Scale the reads by flooring
sum_scale_floor <- scale_reads(physeq = sum_reps, n = min(sample_sums(sum_reps)), round = "floor")

##  Histogram of Sample Read Counts 
sum_scale_floor_plot <- ggplot(data.frame(sum=sample_sums(sum_scale_floor)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="thistle1", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Sum & Scale (by flooring) the Reads"), atop("Binwidth = 20"), ""))) 

# Step 3:  Scale the reads by rounding
sum_scale_round <- scale_reads(physeq = sum_reps, n = min(sample_sums(sum_reps)), round = "round")

##  Histogram of Sample Read Counts 
sum_scale_round_plot <- ggplot(data.frame(sum=sample_sums(sum_scale_round)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="thistle1", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Sum & Scale (by 'R' rounding) the Reads"), atop("Binwidth = 20"), ""))) 

# Step 3:  Scale the reads 
sum_scale_matround <- scale_reads(physeq = sum_reps, n = min(sample_sums(sum_reps)), round = "matround")


data_dup_sum_scale_matround <- read.table("~/Final_PAFL_Trophicstate/raw_data/duplicate_metadata.txt", header = TRUE, sep = "\t")
row.names(data_dup_sum_scale_matround) <- data_dup_sum_scale_matround$names
data_dup_sum_scale_matround$quadrant <- factor(data_dup_sum_scale_matround$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
# Add a column of Total Sequence sums 
TotalSeqs_sum_reps <- data.frame(rowSums(otu_table(sum_reps)))
TotalSeqs_sum_reps$names <- as.factor(row.names(TotalSeqs_sum_reps))
colnames(TotalSeqs_sum_reps)[1] <- "TotalSeqs_Summed"
data_dup_sum_scale_matround_join<- left_join(data_dup_sum_scale_matround, TotalSeqs_sum_reps, by = "names") 

TotalSeqs_sum_scale_matround <- data.frame(rowSums(otu_table(sum_scale_matround)))
TotalSeqs_sum_scale_matround$names <- as.factor(row.names(TotalSeqs_sum_scale_matround))
colnames(TotalSeqs_sum_scale_matround)[1] <- "TotalSeqs_Scaled"
data_dup_sum_scale_matround_join<- left_join(data_dup_sum_scale_matround_join, TotalSeqs_sum_scale_matround, by = "names") 
row.names(data_dup_sum_scale_matround_join) <- data_dup_sum_scale_matround_join$names

data_dup_sum_scale_matround_join <- sample_data(data_dup_sum_scale_matround_join)  # Metadata in our phyloseq object
sum_scale_matround_otu <- otu_table(sum_scale_matround)  # OTU table in our phyloseq object
sum_scale_matround_tax <- tax_table(sum_scale_matround)  # Taxonomy Table in our phyloseq object
sum_scale_matround <- merge_phyloseq(sum_scale_matround_otu, sum_scale_matround_tax, data_dup_sum_scale_matround_join) #####  THIS IS OUR RAW MERGED SAMPLES PHYLOSEQ OBJECT.


##  Histogram of Sample Read Counts 
sum_scale_matround_plot <- ggplot(data.frame(sum=sample_sums(sum_scale_matround)),aes(sum)) + ylab("Number of Sequences per Sample") +
  geom_histogram(colour="black",fill="thistle1", binwidth = 20)  + xlab("Total Sequences") + 
  ggtitle(expression(atop(bold("Sum & Scale (by 'True' rounding) the Reads"), atop("Binwidth = 20"), ""))) 
```


```{r Sequencing Depth Table, results = 'asis', fig.width = 15, fig.height = 6}
manipulations <- c("Raw Samples", "Mean Samples Without Scaling", "Mean and 'R' Rounded Samples Without Scaling", "Mean and 'True' Rounded Samples Without Scaling", "Mean and Floored Samples Without Scaling", "Mean Samples with Scaling (and flooring) the Reads", "Mean Samples with Scaling ('R' rounding) the Reads", "Mean Samples with Scaling ('True' rounding) the Reads", "Sum & Scale (by flooring) the Reads", "Sum & Scale (by 'R' rounding) the Reads", "Sum & Scale (by 'True' rounding) the Reads")

mins <- c(min(sample_sums(bact_samples)), min(sample_sums(raw_merged)),  min(sample_sums(round_mean_merge)), min(sample_sums(matround_mean_merge)),min(sample_sums(floor_mean_merge)), min(sample_sums(scaled_merged_floor)), min(sample_sums(scaled_merged_round)), min(sample_sums(scaled_merged_matround)), min(sample_sums(sum_scale_floor)), min(sample_sums(sum_scale_round)), min(sample_sums(sum_scale_matround)))

means <- c(round(mean(sample_sums(bact_samples)), digits = 2), round(mean(sample_sums(raw_merged)), digits = 2), round(mean(sample_sums(round_mean_merge)), digits = 2),round(mean(sample_sums(matround_mean_merge)), digits = 2), round(mean(sample_sums(floor_mean_merge)), digits = 2), round(mean(sample_sums(scaled_merged_floor)), digits = 2), round(mean(sample_sums(scaled_merged_round)), digits = 2), round(mean(sample_sums(scaled_merged_matround)), digits = 2), round(mean(sample_sums(sum_scale_floor)), digits = 2), round(mean(sample_sums(sum_scale_round)), digits = 2), round(mean(sample_sums(sum_scale_matround)), digits = 2))

medians <- c(median(sample_sums(bact_samples)), median(sample_sums(raw_merged)), median(sample_sums(round_mean_merge)), median(sample_sums(matround_mean_merge)), median(sample_sums(floor_mean_merge)), median(sample_sums(scaled_merged_floor)),median(sample_sums(scaled_merged_round)), median(sample_sums(scaled_merged_matround)), median(sample_sums(sum_scale_floor)), median(sample_sums(sum_scale_round)), median(sample_sums(sum_scale_matround)))

maxs <- c(max(sample_sums(bact_samples)), max(sample_sums(raw_merged)), max(sample_sums(round_mean_merge)), max(sample_sums(matround_mean_merge)), max(sample_sums(floor_mean_merge)), max(sample_sums(scaled_merged_floor)), max(sample_sums(scaled_merged_round)), max(sample_sums(scaled_merged_matround)), max(sample_sums(sum_scale_floor)), max(sample_sums(sum_scale_round)), max(sample_sums(sum_scale_matround)))

ranges <- c(max(sample_sums(bact_samples)) - min(sample_sums(bact_samples)), max(sample_sums(raw_merged)) - min(sample_sums(raw_merged)), max(sample_sums(round_mean_merge)) - min(sample_sums(round_mean_merge)), max(sample_sums(matround_mean_merge)) - min(sample_sums(matround_mean_merge)), max(sample_sums(floor_mean_merge)) - min(sample_sums(floor_mean_merge)), max(sample_sums(scaled_merged_floor)) - min(sample_sums(scaled_merged_floor)), max(sample_sums(scaled_merged_round)) - min(sample_sums(scaled_merged_round)), max(sample_sums(scaled_merged_matround)) - min(sample_sums(scaled_merged_matround)), max(sample_sums(sum_scale_floor)) - min(sample_sums(sum_scale_floor)), max(sample_sums(sum_scale_round)) - min(sample_sums(sum_scale_round)), max(sample_sums(sum_scale_matround)) - min(sample_sums(sum_scale_matround)))

num_otus <- c(nrow(otu_table(bact_samples)), nrow(otu_table(raw_merged)), nrow(otu_table(round_mean_merge)), nrow(otu_table(matround_mean_merge)), nrow(otu_table(floor_mean_merge)), nrow(otu_table(scaled_merged_floor)), nrow(otu_table(scaled_merged_round)), nrow(otu_table(scaled_merged_matround)),  ncol(otu_table(sum_scale_floor)), ncol(otu_table(sum_scale_round)), ncol(otu_table(sum_scale_matround)))

# Make the data frame 
seq_depth <- data.frame(matrix(ncol = 0, nrow = 11))
row.names(seq_depth) <- manipulations
seq_depth$Minimum_Seqs <- mins
seq_depth$Mean_Seqs <- means
seq_depth$Median_Seqs <- medians
seq_depth$Maximum_Seqs <- maxs
seq_depth$Range_Seqs <- ranges
seq_depth$Number_of_OTUs <- num_otus
```

## Meaning between replicates without scaling
```{r}
###  Time to make the pander output of our table!  
panderOptions("digits", 1) # This will set the values to numbers and then will round to the whole number
set.alignment('center', row.names = 'right') # This will align all cells with "center" alignment and the row.names with "right" alignment
pander(seq_depth[1:5,], caption="Table A:  Simple statistics on the sequencing depth while merging replicates WITHOUT scaling.")

```

## **Mean** between replicates and then scale the sample read counts
```{r mean scale stats, fig.width = 15, fig.height = 6}
multiplot(mean_scaled_floor_plot, mean_scaled_round_plot, mean_scaled_matround_plot,cols = 3)

panderOptions("digits", 1) # This will set the values to numbers and then will round to the whole number
set.alignment('center', row.names = 'right') # This will align all cells with "center" alignment and the row.names with "right" alignment
pander(seq_depth[6:8,], caption="Table A:  Simple statistics on the sequencing depth while merging replicates WITH scaling.")
```

## **Sum** between replicates and then scale the sample read counts
```{r sum-scaled stats, fig.width = 15, fig.height = 6}
multiplot(sum_scale_floor_plot, sum_scale_round_plot, sum_scale_matround_plot,cols = 3)

panderOptions("digits", 1) # This will set the values to numbers and then will round to the whole number
set.alignment('center', row.names = 'right') # This will align all cells with "center" alignment and the row.names with "right" alignment
pander(seq_depth[9:11,], caption="Table A:  Simple statistics on the sequencing depth while SUMMING replicates WITH scaling.")
```



```{r Create phyloseq object}
###  Time to create our phyloseq object with the merged and scaled sample reads.
merged_final <- scaled_merged_round

########## ADD THE PROTEOBACTERIA TO THE PHYLA
phy <- data.frame(tax_table(merged_final))
Phylum <- as.character(phy$Phylum)
Class <- as.character(phy$Class)

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Proteobacteria"){
    Phylum[i] <- Class[i]
  } 
}

phy$Phylum <- Phylum
t <- tax_table(as.matrix(phy))

tax_table(merged_final) <- t

# Our phyloseq object!
#merged_final 

## The same object without the 2 wintergreen "hypolimnion" samples
nowin_final <- subset_samples(merged_final, names != "WINH" & names != "WINH3um")
nowin_final <- prune_taxa(taxa_sums(nowin_final) > 0, nowin_final)

```


```{r Alpha Diversity - rarefied, fig.height=10, fig.width=16, eval = FALSE}
####################################################  WITHIN-SAMPLE (ALPHA) DIVERSITY  ####################################################  Working up to Figure 1
####################################################  WITHIN-SAMPLE (ALPHA) DIVERSITY  ####################################################  Working up to Figure 1
####################################################  WITHIN-SAMPLE (ALPHA) DIVERSITY  ####################################################  Working up to Figure 1
# For alpha diversity we will RAREFY our phyloseq object.
#  We will use the mean AND floored data.  
#floor_mean_merge  # Raw Merged Samples, we have 9,854 OTUs  
#raw_nowin <- subset_samples(floor_mean_merge, names != "WINH" & names != "WINH3um")
#raw_nowin <- prune_taxa(taxa_sums(raw_nowin) > 0, raw_nowin)
# Now, we have 9,740 OTUs !

#Data Import for rarefied data 
#raw_nowin  
#raw_merged_min_minus1 <- min(sample_sums(raw_nowin)) - 1

# Following code from Michelle's Butterflygut website: http://rstudio-pubs-static.s3.amazonaws.com/25722_9fc9bdc48f0d4f13b05fa61baeda57a0.html#alpha-diversity
# Rarefy to 14937 reads with replacement 100 times to estimate species richness
# Since we are rarefying to 14937 reads we need to remove the two samples with less than 1000 reads
#raredata14854 <- prune_samples(sample_sums(raw_nowin) > raw_merged_min_minus1, raw_nowin)

# Initialize matrices to store richness and evenness estimates
#richness <- matrix(nrow = 41,ncol = 100)  # Store richness:  We have 41 samples 
#row.names(richness) <- sample_names(raredata14854)
#evenness <- matrix(nrow = 41,ncol = 100)  #Store evenness
#row.names(evenness) <- sample_names(raredata14854)

# We want to be reproducible - so let's set the seed.
#set.seed(15879966) # Same seed as in the first chunk

# For 100 replications, rarefy the OTU table to 14854 reads and store the richness and evennes estimates in our 2 matrices we just created.
#The default for the rarefy_even_depth command is to pick with replacement so I set it to false. Picking without replacement is more computationally intensive 
#for (i in 1:100) {
#  r <- rarefy_even_depth(raredata14854, sample.size = raw_merged_min_minus1, verbose = FALSE, replace = FALSE)
#  rich <- as.numeric(as.matrix(estimate_richness(r, measures="Observed")))
#  richness[,i] <- rich
#  even <- as.numeric(as.matrix(estimate_richness(r, measures="InvSimpson")))
#  evenness[,i] <- even
#}

#write.table(richness, "~/Final_PAFL_Trophicstate/alpha_data/ObservedRichness100", row.names = TRUE)
#write.table(evenness, "~/Final_PAFL_Trophicstate/alpha_data/InvSimpson100", row.names = TRUE)

richness <- read.table("~/Final_PAFL_Trophicstate/alpha_data/ObservedRichness100", header = TRUE)
evenness <- read.table("~/Final_PAFL_Trophicstate/alpha_data/InvSimpson100", header = TRUE)

# Create a new matrix to hold the means and standard deviations of all the richness estimates
rich_stats = matrix(nrow= nrow(richness), ncol=1)
rich_stats[,1] = apply(richness, 1, mean)
#rich_stats[,2] = apply(richness, 1, sd)
rich_stats = data.frame(row.names(richness), rich_stats)
colnames(rich_stats) = c("samples","mean_value")
rich_stats$Test <- "Observed Richness"

# Create a new matrix to hold the means and standard deviations of the evenness estimates
even_stats = matrix(nrow = nrow(evenness), ncol = 1)
even_stats[,1] = apply(evenness, 1, mean)
#even_stats[,2] = apply(evenness, 1, sd)
even_stats = data.frame(row.names(evenness), even_stats)
colnames(even_stats) = c("samples","mean_value")
even_stats$Test <- "Inverse Simpson"


###  Add SIMPSON'S EVENNESS!
#simps_even = as.data.frame(matrix(nrow = nrow(evenness), ncol = 2))
simps_even <- data.frame(matrix(nrow = nrow(evenness), ncol = 3))
colnames(simps_even) = c("samples","mean_value", "Test")
simps_even$samples <- even_stats$samples
simps_even$mean_value <-  even_stats$mean_value/rich_stats$mean_value
simps_even$Test <- "Simpson's Evenness"

#Combine the rich.stats and the even.stats dataframes
alpha_stats <- rbind(rich_stats, even_stats, simps_even)
alpha_stats$names <- alpha_stats$samples
alpha_stats <- makeCategories_dups(alpha_stats)
alpha_stats$ProdLevel <- as.character(alpha_stats$trophicstate)
alpha_stats$ProdLevel[alpha_stats$trophicstate == "Eutrophic"] <- "Productive"
alpha_stats$ProdLevel[alpha_stats$trophicstate == "Mesotrophic"] <- "Productive"
alpha_stats$ProdLevel[alpha_stats$trophicstate == "Oligotrophic"] <- "Unproductive"
alpha_stats$ProdLevel[alpha_stats$lakenames == "Sherman"] <- "Mixed"
alpha_stats$trophicstate[alpha_stats$lakenames == "Sherman"] <- "Mixed"


####  Add all information together!
for(i in 1:length(alpha_stats$limnion)){
  alpha_stats$troph_lim[i]<-paste(as.character(alpha_stats$trophicstate[i]),
                                  as.character(alpha_stats$limnion[i]),
                                  as.character(alpha_stats$filter[i]))}


stats <- alpha_stats %>%
  group_by(troph_lim, Test) %>%
  summarize(mean = mean(mean_value), 
            SE = se(mean_value), 
            SD = sd(mean_value), 
            median = median(mean_value))

final_alpha_stats <- left_join(alpha_stats, stats, by = c("troph_lim", "Test"))


final_alpha_stats$trophicstate <-factor(final_alpha_stats$trophicstate,levels=c("Eutrophic", "Mesotrophic", "Oligotrophic", "Mixed"))
final_alpha_stats$troph_lim <-factor(final_alpha_stats$troph_lim,levels=c("Eutrophic Epilimnion Particle", "Eutrophic Epilimnion Free", "Eutrophic Hypolimnion Particle", "Eutrophic Hypolimnion Free",
                                                              "Mesotrophic Epilimnion Particle", "Mesotrophic Epilimnion Free", "Mesotrophic Hypolimnion Particle", "Mesotrophic Hypolimnion Free",
                                                              "Oligotrophic Epilimnion Particle", "Oligotrophic Epilimnion Free", "Oligotrophic Hypolimnion Particle", "Oligotrophic Hypolimnion Free",
                                                              "Mixed Mixed Particle", "Mixed Mixed Free"))



troph_alpha_plot <- ggplot(final_alpha_stats, aes(x = troph_lim, y = mean_value, color = troph_lim)) + geom_point(size = 5, alpha = 0.7) +
  ggtitle("Alpha Diversity: Rarefied at 14,854") + theme_bw() +   xlab("Habitat") + ylab("") + 
  geom_errorbar(aes(ymin= mean-SD, ymax=mean+SD), color = "black", width=.1, position=position_dodge(.9)) +
  layer(data = final_alpha_stats, mapping = aes(x = troph_lim, y = median), geom = "point", pch = 95, size = 10, color = "black") + 
  layer(data = final_alpha_stats, mapping = aes(x = troph_lim, y = mean), geom = "point", pch = 126, size = 10, color = "red") + 
  facet_grid(Test ~ trophicstate, scales="free", space="free_x") +
    scale_color_manual(name = "", limits=c("Eutrophic Epilimnion Particle", "Eutrophic Epilimnion Free", "Eutrophic Hypolimnion Particle", "Eutrophic Hypolimnion Free",
                                         "Mesotrophic Epilimnion Particle", "Mesotrophic Epilimnion Free", "Mesotrophic Hypolimnion Particle", "Mesotrophic Hypolimnion Free",
                                         "Oligotrophic Epilimnion Particle", "Oligotrophic Epilimnion Free", "Oligotrophic Hypolimnion Particle", "Oligotrophic Hypolimnion Free",
                                         "Mixed Mixed Particle", "Mixed Mixed Free"), 
                     values = c("deeppink", "deeppink", "deeppink", "deeppink","orange","orange","orange","orange", 
                                "turquoise3","turquoise3","turquoise3","turquoise3", "blue", "blue"))+
  scale_x_discrete(breaks=c("Eutrophic Epilimnion Particle", "Eutrophic Epilimnion Free", "Eutrophic Hypolimnion Particle", "Eutrophic Hypolimnion Free",
                            "Mesotrophic Epilimnion Particle", "Mesotrophic Epilimnion Free", "Mesotrophic Hypolimnion Particle", "Mesotrophic Hypolimnion Free",
                            "Oligotrophic Epilimnion Particle", "Oligotrophic Epilimnion Free", "Oligotrophic Hypolimnion Particle", "Oligotrophic Hypolimnion Free",
                            "Mixed Mixed Particle", "Mixed Mixed Free"), 
                   labels=c("Epilimnion Particle", "Epilimnion Free", "Hypolimnion Particle", "Hypolimnion Free",
                            "Epilimnion Particle", "Epilimnion Free", "Hypolimnion Particle", "Hypolimnion Free",
                            "Epilimnion Particle", "Epilimnion Free", "Hypolimnion Particle", "Hypolimnion Free",
                            "Particle", "Free")) + 
  theme(axis.title.x = element_text(face="bold", size=16),
        axis.text.x = element_text(angle=30, colour = "black", vjust=1, hjust = 1, size=14),
        axis.text.y = element_text(colour = "black", size=14),
        axis.title.y = element_text(face="bold", size=16),
        plot.title = element_text(face="bold", size = 20),
        strip.text.x = element_text(size=12, face="bold"),
        strip.text.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size = 14),
        strip.background = element_blank(),
        legend.position="none")



#######  By productivity

####  Add all information together!
for(i in 1:length(alpha_stats$limnion)){
  alpha_stats$prod_lim[i]<-paste(as.character(alpha_stats$ProdLevel[i]),
                                  as.character(alpha_stats$limnion[i]),
                                  as.character(alpha_stats$filter[i]))}


prod_stats <- alpha_stats %>%
  group_by(prod_lim, Test) %>%
  summarize(mean = mean(mean_value), 
            SE = se(mean_value), 
            SD = sd(mean_value), 
            median = median(mean_value))

final_prod_stats <- left_join(alpha_stats, prod_stats, by = c("prod_lim", "Test"))


final_prod_stats$ProdLevel <-factor(final_prod_stats$ProdLevel,levels=c("Productive", "Unproductive", "Mixed"))
final_prod_stats$prod_lim <-factor(final_prod_stats$prod_lim,levels=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                                                      "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                                                      "Mixed Mixed Particle", "Mixed Mixed Free"))

prod_alpha_plot <- ggplot(final_prod_stats, aes(x = prod_lim, y = mean_value, color = prod_lim)) + geom_point(size = 5, alpha = 0.7) +
  ggtitle("Alpha Diversity:  Rarefied at 14,854") + theme_bw() +   xlab("Habitat") + ylab("") + 
  geom_errorbar(aes(ymin= mean-SD, ymax=mean+SD), color = "black", width=.1, position=position_dodge(.9)) +
  layer(data = final_prod_stats, mapping = aes(x = prod_lim, y = median), geom = "point", pch = 95, size = 10, color = "black") + 
  layer(data = final_prod_stats, mapping = aes(x = prod_lim, y = mean), geom = "point", pch = 126, size = 10, color = "red") + 
  facet_grid(Test ~ ProdLevel, scales="free", space="free_x") +
  scale_color_manual(name = "", limits=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                         "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                         "Mixed Mixed Particle", "Mixed Mixed Free"), 
                     values = c("deeppink", "deeppink", "deeppink", "deeppink",
                                "turquoise3","turquoise3","turquoise3","turquoise3", "blue", "blue"))+
  scale_x_discrete(breaks=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                            "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                            "Mixed Mixed Particle", "Mixed Mixed Free"), 
                   labels=c("Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Particle", "Free")) + 
  theme(axis.title.x = element_text(face="bold", size=16),
        axis.text.x = element_text(angle=30, colour = "black", vjust=1, hjust = 1, size=14),
        axis.text.y = element_text(colour = "black", size=14),
        axis.title.y = element_text(face="bold", size=16),
        plot.title = element_text(face="bold", size = 20),
        strip.text.x = element_text(size=12, face="bold"),
        strip.text.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size = 14),
        strip.background = element_blank(),
        legend.position="none")




multiplot(troph_alpha_plot, prod_alpha_plot, cols = 2)
```

## Influence of Rarefying and Scaling the Reads on Alpha Diversity
```{r Alpha diversity - scaled, eval = FALSE}

###  Using the floored data 
##  First remove wintergreen "hypolimnion" samples 
nowin_scaled_merged_floor <- subset_samples(scaled_merged_floor, names != "WINH" & names != "WINH3um")
nowin_scaled_merged_floor <- prune_taxa(taxa_sums(nowin_scaled_merged_floor) > 0, nowin_scaled_merged_floor)

# Get sample names and calculate Observed richness and Inverse Simpson
names <- sample_names(nowin_scaled_merged_floor)
scaled_floor_rich <- as.numeric(as.matrix(estimate_richness(nowin_scaled_merged_floor, measures="Observed")))
scaled_floor_even <- as.numeric(as.matrix(estimate_richness(nowin_scaled_merged_floor, measures="InvSimpson")))
scaled_floor_simpseven <- scaled_floor_even/scaled_floor_rich

#Create new data frame
scaled_floor_alpha <- data.frame(matrix(ncol = 0, nrow = length(names)))
scaled_floor_alpha$names <- names
scaled_floor_alpha$ObsRichness <- scaled_floor_rich
scaled_floor_alpha$InvSimpson <- scaled_floor_even
scaled_floor_alpha$Simps_Even <- scaled_floor_simpseven

# Reshape the data
scaled_floor_alpha <- gather(scaled_floor_alpha, "Test", "Value", 2:4)
scaled_floor_alpha <- makeCategories_dups(scaled_floor_alpha)
scaled_floor_alpha$ProdLevel <- as.character(scaled_floor_alpha$trophicstate)
scaled_floor_alpha$ProdLevel[scaled_floor_alpha$trophicstate == "Eutrophic"] <- "Productive"
scaled_floor_alpha$ProdLevel[scaled_floor_alpha$trophicstate == "Mesotrophic"] <- "Productive"
scaled_floor_alpha$ProdLevel[scaled_floor_alpha$trophicstate == "Oligotrophic"] <- "Unproductive"
scaled_floor_alpha$ProdLevel[scaled_floor_alpha$lakenames == "Sherman"] <- "Mixed"
scaled_floor_alpha$trophicstate[scaled_floor_alpha$lakenames == "Sherman"] <- "Mixed"
# Change test 
scaled_floor_alpha$Test <- as.character(scaled_floor_alpha$Test)
scaled_floor_alpha$Test[scaled_floor_alpha$Test == "ObsRichness"] <- "Observed Richness"
scaled_floor_alpha$Test[scaled_floor_alpha$Test == "InvSimpson"] <- "Inverse Simpson"
scaled_floor_alpha$Test[scaled_floor_alpha$Test == "Simps_Even"] <- "Simpson's Evenness"

for(i in 1:length(scaled_floor_alpha$limnion)){
  scaled_floor_alpha$prod_lim[i]<-paste(as.character(scaled_floor_alpha$ProdLevel[i]),
                                  as.character(scaled_floor_alpha$limnion[i]),
                                  as.character(scaled_floor_alpha$filter[i]))}

# Calculate summary statistics 
scaled_floor_prod_stats <- scaled_floor_alpha %>%
  group_by(prod_lim, Test) %>%
  summarize(mean = mean(Value), 
            SE = se(Value), 
            SD = sd(Value), 
            median = median(Value))


final_scaled_floor_prod_stats <- left_join(scaled_floor_alpha, scaled_floor_prod_stats, by = c("prod_lim", "Test"))

final_scaled_floor_prod_stats$ProdLevel <-factor(final_scaled_floor_prod_stats$ProdLevel,levels=c("Productive", "Unproductive", "Mixed"))
final_scaled_floor_prod_stats$prod_lim <-factor(final_scaled_floor_prod_stats$prod_lim,levels=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                                                      "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                                                      "Mixed Mixed Particle", "Mixed Mixed Free"))

scaled_floor_prod_alpha_plot <- ggplot(final_scaled_floor_prod_stats, aes(x = prod_lim, y = Value, color = prod_lim)) + geom_point(size = 5, alpha = 0.7) +
  ggtitle("Alpha Diversity: Scaled and Floored") + theme_bw() +   xlab("Habitat") + ylab("") + 
  geom_errorbar(aes(ymin= mean-SD, ymax=mean+SD), color = "black", width=.1, position=position_dodge(.9)) +
  layer(data = final_scaled_floor_prod_stats, mapping = aes(x = prod_lim, y = median), geom = "point", pch = 95, size = 10, color = "black") + 
  layer(data = final_scaled_floor_prod_stats, mapping = aes(x = prod_lim, y = mean), geom = "point", pch = 126, size = 10, color = "red") + 
  facet_grid(Test ~ ProdLevel, scales="free", space="free_x") +
  scale_color_manual(name = "", limits=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                         "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                         "Mixed Mixed Particle", "Mixed Mixed Free"), 
                     values = c("deeppink", "deeppink", "deeppink", "deeppink",
                                "turquoise3","turquoise3","turquoise3","turquoise3", "blue", "blue"))+
  scale_x_discrete(breaks=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                            "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                            "Mixed Mixed Particle", "Mixed Mixed Free"), 
                   labels=c("Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Particle", "Free")) + 
  theme(axis.title.x = element_text(face="bold", size=16),
        axis.text.x = element_text(angle=30, colour = "black", vjust=1, hjust = 1, size=14),
        axis.text.y = element_text(colour = "black", size=14),
        axis.title.y = element_text(face="bold", size=16),
        plot.title = element_text(face="bold", size = 20),
        strip.text.x = element_text(size=12, face="bold"),
        strip.text.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size = 14),
        strip.background = element_blank(),
        legend.position="none")


### Using the rounded data
# Get sample names and calculate Observed richness and Inverse Simpson
names <- sample_names(nowin_final)
scaled_round_rich <- as.numeric(as.matrix(estimate_richness(nowin_final, measures="Observed")))
scaled_round_even <- as.numeric(as.matrix(estimate_richness(nowin_final, measures="InvSimpson")))
scaled_round_simpseven <- scaled_round_even/scaled_round_rich

#Create new data frame
scaled_round_alpha <- data.frame(matrix(ncol = 0, nrow = length(names)))
scaled_round_alpha$names <- names
scaled_round_alpha$ObsRichness <- scaled_round_rich
scaled_round_alpha$InvSimpson <- scaled_round_even
scaled_round_alpha$Simps_Even <- scaled_round_simpseven

# Reshape the data
scaled_round_alpha <- gather(scaled_round_alpha, "Test", "Value", 2:4)
scaled_round_alpha <- makeCategories_dups(scaled_round_alpha)
scaled_round_alpha$ProdLevel <- as.character(scaled_round_alpha$trophicstate)
scaled_round_alpha$ProdLevel[scaled_round_alpha$trophicstate == "Eutrophic"] <- "Productive"
scaled_round_alpha$ProdLevel[scaled_round_alpha$trophicstate == "Mesotrophic"] <- "Productive"
scaled_round_alpha$ProdLevel[scaled_round_alpha$trophicstate == "Oligotrophic"] <- "Unproductive"
scaled_round_alpha$ProdLevel[scaled_round_alpha$lakenames == "Sherman"] <- "Mixed"
scaled_round_alpha$trophicstate[scaled_round_alpha$lakenames == "Sherman"] <- "Mixed"
# Change test 
scaled_round_alpha$Test <- as.character(scaled_round_alpha$Test)
scaled_round_alpha$Test[scaled_round_alpha$Test == "ObsRichness"] <- "Observed Richness"
scaled_round_alpha$Test[scaled_round_alpha$Test == "InvSimpson"] <- "Inverse Simpson"
scaled_round_alpha$Test[scaled_round_alpha$Test == "Simps_Even"] <- "Simpson's Evenness"

for(i in 1:length(scaled_round_alpha$limnion)){
  scaled_round_alpha$prod_lim[i]<-paste(as.character(scaled_round_alpha$ProdLevel[i]),
                                  as.character(scaled_round_alpha$limnion[i]),
                                  as.character(scaled_round_alpha$filter[i]))}

# Calculate summary statistics 
scaled_round_prod_stats <- scaled_round_alpha %>%
  group_by(prod_lim, Test) %>%
  summarize(mean = mean(Value), 
            SE = se(Value), 
            SD = sd(Value), 
            median = median(Value))



final_scaled_round_prod_stats <- left_join(scaled_round_alpha, scaled_round_prod_stats, by = c("prod_lim", "Test"))

final_scaled_round_prod_stats$ProdLevel <-factor(final_scaled_round_prod_stats$ProdLevel,levels=c("Productive", "Unproductive", "Mixed"))
final_scaled_round_prod_stats$prod_lim <-factor(final_scaled_round_prod_stats$prod_lim,levels=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                                                      "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                                                      "Mixed Mixed Particle", "Mixed Mixed Free"))

scaled_round_prod_alpha_plot <- ggplot(final_scaled_round_prod_stats, aes(x = prod_lim, y = Value, color = prod_lim)) + geom_point(size = 5, alpha = 0.7) +
  ggtitle("Alpha Diversity: Scaled and Rounded") + theme_bw() +   xlab("Habitat") + ylab("") + 
  #scale_y_continuous(breaks = seq(0,1500, 250)) + 
  geom_errorbar(aes(ymin= mean-SD, ymax=mean+SD), color = "black", width=.1, position=position_dodge(.9)) +
  layer(data = final_scaled_round_prod_stats, mapping = aes(x = prod_lim, y = median), geom = "point", pch = 95, size = 10, color = "black") + 
  layer(data = final_scaled_round_prod_stats, mapping = aes(x = prod_lim, y = mean), geom = "point", pch = 126, size = 10, color = "red") + 
  facet_grid(Test ~ ProdLevel, scales="free", space="free_x") +
  scale_color_manual(name = "", limits=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                         "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                         "Mixed Mixed Particle", "Mixed Mixed Free"), 
                     values = c("deeppink", "deeppink", "deeppink", "deeppink",
                                "turquoise3","turquoise3","turquoise3","turquoise3", "blue", "blue"))+
  scale_x_discrete(breaks=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                            "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                            "Mixed Mixed Particle", "Mixed Mixed Free"), 
                   labels=c("Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Particle", "Free")) + 
  theme(axis.title.x = element_text(face="bold", size=16),
        axis.text.x = element_text(angle=30, colour = "black", vjust=1, hjust = 1, size=14),
        axis.text.y = element_text(colour = "black", size=14),
        axis.title.y = element_text(face="bold", size=16),
        plot.title = element_text(face="bold", size = 20),
        strip.text.x = element_text(size=12, face="bold"),
        strip.text.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size = 14),
        strip.background = element_blank(),
        legend.position="none")

```


```{r comparing alpha diversity methods, fig.width = 28, fig.height = 20, eval = FALSE, echo = FALSE}
multiplot(prod_alpha_plot, scaled_round_prod_alpha_plot, scaled_floor_prod_alpha_plot, sum_scaled_matround_prod_alpha_plot, cols =2)

# Create a data frame to visually compare all the values 
# Rarefied data 
raw_alpha2 <- prod_stats %>%
  select(Test, mean)
colnames(raw_alpha2)[3] <- "mean_rarefied"

# Scaled and Floored data 
scale_floor_alpha2 <- scaled_floor_prod_stats %>%
  select(Test, mean)
colnames(scale_floor_alpha2)[3] <- "mean_scaled_floor"

# Scaled and rounded data 
scale_round_alpha2 <- scaled_round_prod_stats %>%
  select(Test, mean)
colnames(scale_round_alpha2)[3] <- "mean_scaled_round"

# Sum, Scale and matrounded data 
sum_scale_matround_alpha2 <- sum_scaled_matround_prod_stats %>%
  select(Test, mean)
colnames(sum_scale_matround_alpha2)[3] <- "sum_scaled_matround"


#Combine the data sets 
joined_alpha<- 
  left_join(raw_alpha2, scale_floor_alpha2, by = c("prod_lim", "Test")) %>%
  left_join(scale_round_alpha2, by = c("prod_lim", "Test")) %>%
  left_join(sum_scale_matround_alpha2, by = c("prod_lim", "Test")) 

# Arrange the data to be sorted by Test
joined_alpha2 <- as.data.frame(joined_alpha) 
compare_alphas <- arrange(joined_alpha2, Test)

###  Time to make the pander output of our table!  
panderOptions("digits", 3) # This will set the values to numbers and then will round to the whole number
set.alignment('center', row.names = 'right') # This will align all cells with "center" alignment and the row.names with "right" alignment
pander(compare_alphas, caption="Table B:  Comparison of Alpha Diversity Measurements.")
```




```{r summed alpha diversity}
# For alpha diversity we will RAREFY our phyloseq object.
#  We will use the mean AND floored data.  
#sum_reps  # Raw Merged Samples, we have 14,378 OTUs  
#sum_reps_nowin <- subset_samples(sum_reps, names != "WINH" & names != "WINH3um")
#sum_reps_nowin <- prune_taxa(taxa_sums(sum_reps_nowin) > 0, sum_reps_nowin)
# Now, we have 14,184 OTUs !

#Data Import for rarefied data 
#sum_reps_nowin  
#sum_reps_nowin_min_minus1 <- min(sample_sums(sum_reps_nowin)) - 1

# Following code from Michelle's Butterflygut website: http://rstudio-pubs-static.s3.amazonaws.com/25722_9fc9bdc48f0d4f13b05fa61baeda57a0.html#alpha-diversity
# Rarefy to 14936 reads with replacement 100 times to estimate species richness
# Since we are rarefying to 14936 reads we need to remove samples with less than 14936 reads
#raredata14936 <- prune_samples(sample_sums(sum_reps_nowin) > sum_reps_nowin_min_minus1, sum_reps_nowin)

# Initialize matrices to store richness and evenness estimates
#richness <- matrix(nrow = 41,ncol = 100)  # Store richness:  We have 41 samples 
#row.names(richness) <- sample_names(raredata14936)
#evenness <- matrix(nrow = 41,ncol = 100)  #Store evenness
#row.names(evenness) <- sample_names(raredata14936)

# We want to be reproducible - so let's set the seed.
#set.seed(15879966) # Same seed as in the first chunk

# For 100 replications, rarefy the OTU table to 14854 reads and store the richness and evennes estimates in our 2 matrices we just created.
#The default for the rarefy_even_depth command is to pick with replacement so I set it to false. Picking without replacement is more computationally intensive 
#for (i in 1:100) {
#  r <- rarefy_even_depth(raredata14936, sample.size = sum_reps_nowin_min_minus1, verbose = FALSE, replace = FALSE)
#  rich <- as.numeric(as.matrix(estimate_richness(r, measures="Observed")))
#  richness[,i] <- rich
#  even <- as.numeric(as.matrix(estimate_richness(r, measures="InvSimpson")))
#  evenness[,i] <- even
#}

#write.table(richness, "~/Final_PAFL_Trophicstate/alpha_data/ObservedRichness100_summed", row.names = TRUE)
#write.table(evenness, "~/Final_PAFL_Trophicstate/alpha_data/InvSimpson100_summed", row.names = TRUE)


richness_summed <- read.table("~/Final_PAFL_Trophicstate/alpha_data/ObservedRichness100_summed", header = TRUE)
evenness_summed <- read.table("~/Final_PAFL_Trophicstate/alpha_data/InvSimpson100_summed", header = TRUE)

# Create a new matrix to hold the means and standard deviations of all the richness_summed estimates
rich_stats_summed <- matrix(nrow= nrow(richness_summed), ncol=1)
rich_stats_summed[,1] <- apply(richness_summed, 1, mean)
#rich_stats_summed[,2] = apply(richness_summed, 1, sd)
rich_stats_summed = data.frame(row.names(richness_summed), rich_stats_summed)
colnames(rich_stats_summed) <- c("samples","mean_value")
rich_stats_summed$Test <- "Observed Richness"

# Create a new matrix to hold the means and standard deviations of the evenness_summed estimates
even_stats_summed <- matrix(nrow = nrow(evenness_summed), ncol = 1)
even_stats_summed[,1] <- apply(evenness_summed, 1, mean)
#even_stats[,2] = apply(evenness_summed, 1, sd)
even_stats_summed <- data.frame(row.names(evenness_summed), even_stats_summed)
colnames(even_stats_summed) <- c("samples","mean_value")
even_stats_summed$Test <- "Inverse Simpson"


###  Add SIMPSON'S EVENNESS!
simps_even_summed <- data.frame(matrix(nrow = nrow(evenness_summed), ncol = 3))
colnames(simps_even_summed) = c("samples","mean_value", "Test")
simps_even_summed$samples <- even_stats_summed$samples
simps_even_summed$mean_value <-  even_stats_summed$mean_value/rich_stats_summed$mean_value
simps_even_summed$Test <- "Simpson's Evenness"


#Combine the rich.stats and the even.stats dataframes
alpha_stats_summed <- rbind(rich_stats_summed, even_stats_summed, simps_even_summed)
alpha_stats_summed$names <- alpha_stats_summed$samples
alpha_stats_summed <- makeCategories_dups(alpha_stats_summed)
alpha_stats_summed$ProdLevel <- as.character(alpha_stats_summed$trophicstate)
alpha_stats_summed$ProdLevel[alpha_stats_summed$trophicstate == "Eutrophic"] <- "Productive"
alpha_stats_summed$ProdLevel[alpha_stats_summed$trophicstate == "Mesotrophic"] <- "Productive"
alpha_stats_summed$ProdLevel[alpha_stats_summed$trophicstate == "Oligotrophic"] <- "Unproductive"
alpha_stats_summed$ProdLevel[alpha_stats_summed$lakenames == "Sherman"] <- "Mixed"
alpha_stats_summed$trophicstate[alpha_stats_summed$lakenames == "Sherman"] <- "Mixed"


####  Add all information together!
for(i in 1:length(alpha_stats_summed$limnion)){
  alpha_stats_summed$prod_lim[i]<-paste(as.character(alpha_stats_summed$ProdLevel[i]),
                                  as.character(alpha_stats_summed$limnion[i]),
                                  as.character(alpha_stats_summed$filter[i]))}


summed_stats <- alpha_stats_summed %>%
  group_by(prod_lim, Test) %>%
  summarize(mean = mean(mean_value), 
            SE = se(mean_value), 
            SD = sd(mean_value), 
            median = median(mean_value))

final_summed_alpha_stats <- left_join(alpha_stats_summed, summed_stats, by = c("prod_lim", "Test"))

final_summed_alpha_stats$ProdLevel <-factor(final_summed_alpha_stats$ProdLevel,levels=c("Productive", "Unproductive", "Mixed"))
final_summed_alpha_stats$prod_lim <-factor(final_summed_alpha_stats$prod_lim,levels=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                                                      "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                                                      "Mixed Mixed Particle", "Mixed Mixed Free"))

final_summed_alpha_stats_plot <- ggplot(final_summed_alpha_stats, aes(x = prod_lim, y = mean_value, color = prod_lim)) + geom_point(size = 5, alpha = 0.7) +
  ggtitle("Alpha Diversity: Summed and Rarefied at 14,936 Sequences") + theme_bw() +   xlab("Habitat") + ylab("") + 
  #scale_y_continuous(breaks = seq(0,1500, 250)) + 
  geom_errorbar(aes(ymin= mean-SD, ymax=mean+SD), color = "black", width=.1, position=position_dodge(.9)) +
  layer(data = final_summed_alpha_stats, mapping = aes(x = prod_lim, y = median), geom = "point", pch = 95, size = 10, color = "black") + 
  layer(data = final_summed_alpha_stats, mapping = aes(x = prod_lim, y = mean), geom = "point", pch = 126, size = 10, color = "red") + 
  facet_grid(Test ~ ProdLevel, scales="free", space="free_x") +
  scale_color_manual(name = "", limits=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                         "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                         "Mixed Mixed Particle", "Mixed Mixed Free"), 
                     values = c("deeppink", "deeppink", "deeppink", "deeppink",
                                "turquoise3","turquoise3","turquoise3","turquoise3", "blue", "blue"))+
  scale_x_discrete(breaks=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                            "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                            "Mixed Mixed Particle", "Mixed Mixed Free"), 
                   labels=c("Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Particle", "Free")) + 
  theme(axis.title.x = element_text(face="bold", size=16),
        axis.text.x = element_text(angle=30, colour = "black", vjust=1, hjust = 1, size=14),
        axis.text.y = element_text(colour = "black", size=14),
        axis.title.y = element_text(face="bold", size=16),
        plot.title = element_text(face="bold", size = 20),
        strip.text.x = element_text(size=12, face="bold"),
        strip.text.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size = 14),
        strip.background = element_blank(),
        legend.position="none")



### Let's check it out!
sum_reps_nowin <- subset_samples(sum_reps, names != "WINH" & names != "WINH3um")
sum_reps_nowin <- prune_taxa(taxa_sums(sum_reps_nowin) > 0, sum_reps_nowin)
nowin_sum_reps <- data.frame(sample_data(sum_reps_nowin))
nowin_sum_reps <- select(nowin_sum_reps, names, TotalSeqs_Summed) 
nowin_sum_reps$names <- as.factor(nowin_sum_reps$names)

sum_reps_alpha_stats <- final_summed_alpha_stats %>%
  filter(Test == "Observed Richness") %>%
  select(names, prod_lim, Test, mean_value, lakenames, filter, quadrant, trophicstate, ProdLevel) %>%
  arrange(names)

sum_reps_alpha_test <- left_join(nowin_sum_reps, sum_reps_alpha_stats, by = "names")

#SOURCE: http://goo.gl/K4yh  and http://stackoverflow.com/posts/7549819/revisions
lm_eqn <- function(df, Y_column, X_column){
    m <- lm(Y_column ~ X_column, df);
    eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2*","~~italic(p)~"="~pval, 
         list(a = format(coef(m)[1], digits = 2), 
              b = format(coef(m)[2], digits = 2), 
             r2 = format(summary(m)$r.squared, digits = 3),
             pval = format(anova(m)$'Pr(>F)'[1], digits = 2)))
    as.character(as.expression(eq));                 
}



ggplot(sum_reps_alpha_test, aes(x = TotalSeqs_Summed, y = mean_value)) + geom_point(size = 5, alpha = 1) + 
  ggtitle("Total Summed Sequences vs Observed Richness") + theme_bw() +   xlab("Total Summed Sequences") + ylab("Observed Richness") + theme(legend.position = "none") +
  geom_smooth(method = "lm") + geom_text(aes(x = 25000, y = 1200, label = lm_eqn(sum_reps_alpha_test, Y_column = mean_value, X_column = TotalSeqs_Summed)), parse = TRUE)

```

```{r sum-scale-matround Alpha diversity, fig.width = 12, fig.height = 15}
### Using the SUM and then SCALED  and MATrounded data
## remove the 2 wintergreen "hypolimnion" samples
nowin_sum_scale_matround <- subset_samples(sum_scale_matround, names != "WINH" & names != "WINH3um")
nowin_sum_scale_matround <- prune_taxa(taxa_sums(nowin_sum_scale_matround) > 0, nowin_sum_scale_matround)

names <- sample_names(nowin_sum_scale_matround)
sum_scaled_matround_rich <- as.numeric(as.matrix(estimate_richness(nowin_sum_scale_matround, measures="Observed")))
sum_scaled_matround_even <- as.numeric(as.matrix(estimate_richness(nowin_sum_scale_matround, measures="InvSimpson")))
sum_scaled_matround_simpseven <- sum_scaled_matround_even/sum_scaled_matround_rich

#Create new data frame
sum_scaled_matround_alpha <- data.frame(matrix(ncol = 0, nrow = length(names)))
sum_scaled_matround_alpha$names <- names
sum_scaled_matround_alpha$ObsRichness <- sum_scaled_matround_rich
sum_scaled_matround_alpha$InvSimpson <- sum_scaled_matround_even
sum_scaled_matround_alpha$Simps_Even <- sum_scaled_matround_simpseven

# Reshape the data
sum_scaled_matround_alpha <- gather(sum_scaled_matround_alpha, "Test", "Value", 2:4)
sum_scaled_matround_alpha <- makeCategories_dups(sum_scaled_matround_alpha)
sum_scaled_matround_alpha$ProdLevel <- as.character(sum_scaled_matround_alpha$trophicstate)
sum_scaled_matround_alpha$ProdLevel[sum_scaled_matround_alpha$trophicstate == "Eutrophic"] <- "Productive"
sum_scaled_matround_alpha$ProdLevel[sum_scaled_matround_alpha$trophicstate == "Mesotrophic"] <- "Productive"
sum_scaled_matround_alpha$ProdLevel[sum_scaled_matround_alpha$trophicstate == "Oligotrophic"] <- "Unproductive"
sum_scaled_matround_alpha$ProdLevel[sum_scaled_matround_alpha$lakenames == "Sherman"] <- "Mixed"
sum_scaled_matround_alpha$trophicstate[sum_scaled_matround_alpha$lakenames == "Sherman"] <- "Mixed"
# Change test 
sum_scaled_matround_alpha$Test <- as.character(sum_scaled_matround_alpha$Test)
sum_scaled_matround_alpha$Test[sum_scaled_matround_alpha$Test == "ObsRichness"] <- "Observed Richness"
sum_scaled_matround_alpha$Test[sum_scaled_matround_alpha$Test == "InvSimpson"] <- "Inverse Simpson"
sum_scaled_matround_alpha$Test[sum_scaled_matround_alpha$Test == "Simps_Even"] <- "Simpson's Evenness"

for(i in 1:length(sum_scaled_matround_alpha$limnion)){
  sum_scaled_matround_alpha$prod_lim[i]<-paste(as.character(sum_scaled_matround_alpha$ProdLevel[i]),
                                  as.character(sum_scaled_matround_alpha$limnion[i]),
                                  as.character(sum_scaled_matround_alpha$filter[i]))}

# Calculate summary statistics 
sum_scaled_matround_prod_stats <- sum_scaled_matround_alpha %>%
  group_by(prod_lim, Test) %>%
  summarize(mean = mean(Value), 
            SE = se(Value), 
            SD = sd(Value), 
            median = median(Value))



final_sum_scaled_matround_prod_stats <- left_join(sum_scaled_matround_alpha, sum_scaled_matround_prod_stats, by = c("prod_lim", "Test"))

final_sum_scaled_matround_prod_stats$ProdLevel <-factor(final_sum_scaled_matround_prod_stats$ProdLevel,levels=c("Productive", "Unproductive", "Mixed"))
final_sum_scaled_matround_prod_stats$prod_lim <-factor(final_sum_scaled_matround_prod_stats$prod_lim,levels=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                                                      "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                                                      "Mixed Mixed Particle", "Mixed Mixed Free"))

sum_scaled_matround_prod_alpha_plot <- ggplot(final_sum_scaled_matround_prod_stats, aes(x = prod_lim, y = Value, color = prod_lim)) + geom_point(size = 5, alpha = 0.7) +
  ggtitle("Alpha Diversity: Sum, Scaled and Rounded") + theme_bw() +   xlab("Habitat") + ylab("") + 
  #scale_y_continuous(breaks = seq(0,1500, 250)) + 
  geom_errorbar(aes(ymin= mean-SD, ymax=mean+SD), color = "black", width=.1, position=position_dodge(.9)) +
  layer(data = final_sum_scaled_matround_prod_stats, mapping = aes(x = prod_lim, y = median), geom = "point", pch = 95, size = 10, color = "black") + 
  layer(data = final_sum_scaled_matround_prod_stats, mapping = aes(x = prod_lim, y = mean), geom = "point", pch = 126, size = 10, color = "red") + 
  facet_grid(Test ~ ProdLevel, scales="free", space="free_x") +
  scale_color_manual(name = "", limits=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                                         "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                                         "Mixed Mixed Particle", "Mixed Mixed Free"), 
                     values = c("deeppink", "deeppink", "deeppink", "deeppink",
                                "turquoise3","turquoise3","turquoise3","turquoise3", "blue", "blue"))+
  scale_x_discrete(breaks=c("Productive Epilimnion Particle", "Productive Epilimnion Free", "Productive Hypolimnion Particle", "Productive Hypolimnion Free",
                            "Unproductive Epilimnion Particle", "Unproductive Epilimnion Free", "Unproductive Hypolimnion Particle", "Unproductive Hypolimnion Free",
                            "Mixed Mixed Particle", "Mixed Mixed Free"), 
                   labels=c("Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Epilimnion \nParticle-Associated", "Epilimnion \nFree-Living", "Hypolimnion \nParticle-Associated", "Hypolimnion \nFree-Living",
                            "Particle", "Free")) + 
  theme(axis.title.x = element_text(face="bold", size=16),
        axis.text.x = element_text(angle=30, colour = "black", vjust=1, hjust = 1, size=14),
        axis.text.y = element_text(colour = "black", size=14),
        axis.title.y = element_text(face="bold", size=16),
        plot.title = element_text(face="bold", size = 20),
        strip.text.x = element_text(size=12, face="bold"),
        strip.text.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size = 14),
        strip.background = element_blank(),
        legend.position="none")


nowin_sum_scaled_data <- data.frame(sample_data(nowin_sum_scale_matround))
nowin_sum_scaled_data_final <- select(nowin_sum_scaled_data, names, TotalSeqs_Summed, TotalSeqs_Scaled) 
nowin_sum_scaled_data_final$names <- as.character(nowin_sum_scaled_data_final$names)

matround_stats_obsrich <- final_sum_scaled_matround_prod_stats %>%
  filter(Test == "Observed Richness") %>%
  select(names, prod_lim, Test, Value, lakenames, filter, quadrant, trophicstate, ProdLevel) %>%
  arrange(names)

alpha_test <- left_join(nowin_sum_scaled_data_final, matround_stats_obsrich, by = "names")



obsrich_vs_summedseqs_plot <- ggplot(alpha_test, aes(x = TotalSeqs_Summed, y = Value)) + geom_point(size = 5, alpha = 1) + 
  ggtitle("Total Summed Sequences vs Observed Richness") + theme_bw() +   xlab("Total Summed Sequences") + ylab("Observed Richness") + 
  scale_y_continuous(breaks = seq(0,1500, 250)) + 
  geom_smooth(method = "lm") + geom_text(aes(x = 25000, y = 1200, label = lm_eqn(alpha_test, Y_column = Value, X_column = TotalSeqs_Summed)), parse = TRUE) + 
  geom_point(aes(x = TotalSeqs_Summed, y = Value, color = prod_lim), size = 5, alpha = 1) + scale_color_brewer(palette="Paired") 
  

obsrich_vs_scaledseqs_plot <- ggplot(alpha_test, aes(x = TotalSeqs_Scaled, y = Value)) + geom_point(size = 5, alpha = 1) + 
  ggtitle("Total Scaled Sequences vs Observed Richness") + theme_bw() +   xlab("Total Scaled Sequences") + ylab("Observed Richness") + 
  scale_y_continuous(breaks = seq(0,1500, 250)) + 
  geom_smooth(method = "lm") + geom_text(aes(x = 14750, y = 1200, label = lm_eqn(alpha_test, Y_column = Value, X_column = TotalSeqs_Scaled)), parse = TRUE) + 
  geom_point(aes(x = TotalSeqs_Scaled, y = Value, color = prod_lim), size = 5, alpha = 1) + scale_color_brewer(palette="Paired") 


multiplot(obsrich_vs_summedseqs_plot, obsrich_vs_scaledseqs_plot, cols = 1)

# Store ggplot graphical parameters
#g1 = ggplotGrob(obsrich_vs_summedseqs_plot)
#g2 = ggplotGrob(obsrich_vs_scaledseqs_plot)

# Add columns to plots without legends
#ncol(g1); ncol(g2)
#g1 <- gtable_add_cols(g1, g2$widths[6])

# Draw everything
#grid.draw(rbind(g1, g2, size = "last"))

```

```{r comparing summed alphas, fig.width = 28, fig.height = 18}
multiplot(final_summed_alpha_stats_plot, sum_scaled_matround_prod_alpha_plot, cols = 2)

# Create a data frame to visually compare all the values 
# Rarefied data 
summed_alph <- summed_stats %>%
  select(Test, mean, prod_lim)
colnames(summed_alph)[2] <- "mean_rarefied"

# Sum, Scale and matrounded data 
summed_scaled_alph <- sum_scaled_matround_prod_stats %>%
  select(Test, mean, prod_lim)
colnames(summed_scaled_alph)[2] <- "mean_summed_scaled"


#Combine the data sets 
joined_summed_alpha <- left_join(summed_scaled_alph, summed_alph, by = c("prod_lim", "Test")) 

# Arrange the data to be sorted by Test
joined_summed_alpha2 <- as.data.frame(joined_summed_alpha) 
compare_summed_alphas <- arrange(joined_summed_alpha2, Test)

###  Time to make the pander output of our table!  
panderOptions("digits", 3) # This will set the values to numbers and then will round to the whole number
set.alignment('center', row.names = 'right') # This will align all cells with "center" alignment and the row.names with "right" alignment
pander(compare_summed_alphas, caption="Table B:  Comparison of Summed & Scaled Alpha Diversity Measurements.")
```




# Figure 2
```{r ordinations, eval = FALSE, echo = FALSE}
####################################################  ORDINATIONS  ####################################################  Working up to Figure 3
####################################################  ORDINATIONS  ####################################################  Working up to FIgure 3
####################################################  ORDINATIONS  ####################################################  Working up to Figure 3
### Clustering 
### Get rid of the Wintergreen HYPOLIMNION Samples
nowinOTU <- otu_table(nowin_final)
#weighted
norm_bray <- vegdist(nowinOTU, method = "bray", binary = FALSE)  # calculates the Bray-Curtis Distances
nowinOTU_df <- data.frame(otu_table(nowin_final))  
norm_soren <- vegdist(nowinOTU_df, method = "bray", binary = TRUE)  # SORENSEN INDEX


########  NMDS + PCoA Plots
#########  PLOT ORDINATIONS FOR BOTH SCALED AND MANUAL
bray_pcoa <- pcoa(norm_bray)
bray_pcoa2 <- bray_pcoa$vectors
bray_pcoa3 <- data.frame(bray_pcoa2[, 1:3])
bray_pcoa3$names <- row.names(bray_pcoa3)
bray_pcoa4 <- makeCategories_dups(bray_pcoa3)
bray_pcoa4$quadrant <- factor(bray_pcoa4$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
### Let's extract the values of each axis for the pcoa
bray_values <- bray_pcoa$values
bray_rel_eigens <- bray_values$Relative_eig
bray_rel_eigen1 <- bray_rel_eigens[1]
bray_rel_eigen1_percent <- round(bray_rel_eigen1 * 100, digits = 1)
bray_rel_eigen2 <- bray_rel_eigens[2]
bray_rel_eigen2_percent <- round(bray_rel_eigen2 * 100, digits = 1)
bray_axis1 <- paste("PCoA1:",bray_rel_eigen1_percent,"%")
bray_axis2 <- paste("PCoA2:",bray_rel_eigen2_percent,"%")

pcoa_bray <- ggplot(bray_pcoa4, aes(Axis.1, Axis.2 * -1, color = quadrant, shape = trophicstate)) +
  xlab(bray_axis1) + ylab(bray_axis2) + #ggtitle("Bray-Curtis: Trophic State") +
  geom_point(size= 6, alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "Habitat", breaks=c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"),
                     labels = c("Free-Living Epilimnion", "Free-Living Mixed",  "Free-Living Hypolimnion", "Particle-Associated Epilimnion", "Particle-Associated Mixed", "Particle-Associated Hypolimnion"), 
                     values = c("purple3", "mediumblue", "darkgreen", "orchid1", "deepskyblue", "green")) +
  scale_shape_manual(name = "Trophic State", breaks = c("Eutrophic", "Mesotrophic", "Oligotrophic", "Mixed"), 
                     labels = c("Eutrophic", "Mesotrophic", "Oligotrophic", "Mixed"),
                     values = c(15, 19, 17)) +
  theme(axis.text.x = element_text(colour="black", vjust=0.5, size=14), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=14),
        axis.title.x = element_text(face="bold", size=16),
        axis.title.y = element_text(face="bold", size=16),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "none"); pcoa_bray



##########  SORENSEN'S DISSIMILARITY
soren_pcoa <- pcoa(norm_soren)
soren_pcoa2 <- soren_pcoa$vectors
soren_pcoa3 <- data.frame(soren_pcoa2[, 1:3])
soren_pcoa3$names <- row.names(soren_pcoa3)
soren_pcoa4 <- makeCategories_dups(soren_pcoa3)
soren_pcoa4$quadrant <- factor(soren_pcoa4$quadrant,levels = c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"))
### Let's extract the values of each axis for the pcoa
soren_values <- soren_pcoa$values
soren_rel_eigens <- soren_values$Relative_eig
soren_rel_eigen1 <- soren_rel_eigens[1]
soren_rel_eigen1_percent <- round(soren_rel_eigen1 * 100, digits = 1)
soren_rel_eigen2 <- soren_rel_eigens[2]
soren_rel_eigen2_percent <- round(soren_rel_eigen2 * 100, digits = 1)
soren_axis1 <- paste("PCoA1:",soren_rel_eigen1_percent,"%")
soren_axis2 <- paste("PCoA2:",soren_rel_eigen2_percent,"%")

pcoa_soren <- ggplot(soren_pcoa4, aes(Axis.1 * -1, Axis.2 * -1, color = quadrant, shape = trophicstate)) +
  xlab(soren_axis1) + ylab(soren_axis2) + #ggtitle("Bray-Curtis: Trophic State") +
  geom_point(size= 6, alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) +
  scale_color_manual(name = "Habitat", breaks=c("Free Epilimnion", "Free Mixed",  "Free Hypolimnion", "Particle Epilimnion", "Particle Mixed", "Particle Hypolimnion"),
                     labels = c("Free-Living Epilimnion", "Free-Living Mixed",  "Free-Living Hypolimnion", "Particle-Associated Epilimnion", "Particle-Associated Mixed", "Particle-Associated Hypolimnion"), 
                     values = c("purple3", "mediumblue", "darkgreen", "orchid1", "deepskyblue", "green")) +
  scale_shape_manual(name = "Trophic State", breaks = c("Eutrophic", "Mesotrophic", "Oligotrophic", "Mixed"), 
                     labels = c("Eutrophic", "Mesotrophic", "Oligotrophic", "Mixed"),
                     values = c(15, 19, 17)) +
  theme(axis.text.x = element_text(colour="black", vjust=0.5, size=14), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=14),
        axis.title.x = element_text(face="bold", size=16),
        axis.title.y = element_text(face="bold", size=16),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "right"); pcoa_soren

#jpeg(filename="~/Final_PAFL_Trophicstate/Figures/Fig.2_NMDS_bc+soren.jpeg", width= 45, height=18, units= "cm", pointsize= 14, res=500)
grid.newpage()
pushViewport(viewport(layout=grid.layout(1,2,width=c(0.4,0.6))))
print(pcoa_bray, vp=viewport(layout.pos.row=1,layout.pos.col=1))
print(pcoa_soren, vp=viewport(layout.pos.row=1,layout.pos.col=2))
#dev.off()


```







## Figure S1:  Depth Profiles between the Lakes 
```{r Figure S1, fig.width=6, fig.height = 6}
####################################################  DEPTH PROFILE  ####################################################  Working up to Figure S1
####################################################  DEPTH PROFILE  ####################################################  Working up to Figure S1
####################################################  DEPTH PROFILE  ####################################################  Working up to Figure S1

profile <- read.csv(file = "~/Final_PAFL_Trophicstate/raw_data/AllLakes_depthprofile.csv", header = TRUE); 
profile <- subset(profile, select = c('lakename',"trophicstate", "depth", "temp", "DO", "pH", "SpC"))
#profile$depth <- as.numeric(profile$depth)
#profile$depth <- as.factor(profile$depth)
profile$DO. = NULL

profile$trophicstate <- as.character(profile$trophicstate)
profile$trophicstate[profile$trophicstate == "Eutrophic"] <- "Productive"
profile$trophicstate[profile$trophicstate == "Mesotrophic"] <- "Productive"
profile$trophicstate[profile$trophicstate == "Oligotrophic"] <- "Unproductive"
profile$trophicstate[profile$trophicstate == "Mixed"] <- "Mixed"

# ALL LAKES 
profile_all <- profile %>%  gather(Variable, Value, 4:7)
profile_all$lakename <- as.character(profile_all$lakename)
profile_all$lakename[profile_all$lakename == "WIN"] <- "Wintergreen"
profile_all$lakename[profile_all$lakename == "SIX"] <- "Sixteen"
profile_all$lakename[profile_all$lakename == "SHE"] <- "Sherman"
profile_all$lakename[profile_all$lakename == "PAY"] <- "Payne"
profile_all$lakename[profile_all$lakename == "LON"] <- "Little Long"
profile_all$lakename[profile_all$lakename == "LEE"] <- "Lee"
profile_all$lakename[profile_all$lakename == "GUL"] <- "Gull"
profile_all$lakename[profile_all$lakename == "BRI"] <- "Bristol"
profile_all$lakename[profile_all$lakename == "BAK"] <- "Baker"
profile_all$lakename[profile_all$lakename == "BAS"] <- "Baseline"
profile_all$lakename[profile_all$lakename == "BST"] <- "Bassett"
profile_all$lakename[profile_all$lakename == "BST"] <- "Bassett"
profile_all$trophicstate[profile_all$trophicstate == "Productive"] <- "High-Nutrient"
profile_all$trophicstate[profile_all$trophicstate == "Unproductive"] <- "Low-Nutrient"
profile_all$trophicstate[profile_all$trophicstate == "Mixed"] <- "High-Nutrient"


profile_all <- subset(profile_all, Variable != "SpC")


profile_labeller <- function(var, value){
  value <- as.character(value)
  if (var=="Variable") { 
    value[value=="temp"] <- "Temperature \n (Celsius)"
    value[value=="DO"]   <- "Dissolved Oxygen \n (mg/L)"
    value[value=="pH"]   <- "pH"
    value[value=="SpC"]   <- "Specific Conductivity \n (uS/cm)"
  }
  return(value)
}

#####  Plotting FIGURE S1  #####  Plotting FIGURE S1  #####  Plotting FIGURE S1  #####  Plotting FIGURE S1  #####  Plotting FIGURE S1
## All LAKES
#tiff(filename="~/Final_PAFL_Trophicstate/Final_Figures/Fig.S1_AllLake_profiles.tiff", width= 30, height=40, units= "cm",pointsize= 18, res=200)
ggplot(profile_all, aes(x=Value, y = depth, color = lakename)) +   
  geom_path(size=1, alpha = 0.8) + ylab("Depth (m)") + xlab("") + 
  theme_bw() +  geom_point(size=2, alpha = 0.8) + geom_hline(h=0) +
  scale_y_reverse(breaks=seq(0, 32, 4), expand = c(0, 0)) + #breaks=seq(0, 30, 5), lim = c(30,0), expand = c(0, 0)
  scale_color_manual(name = "Lake Name", 
                     labels = c("Baker", "Baseline", "Bassett", "Bristol", "Gull", "Lee", "Little Long", "Payne", "Sherman", "Sixteen", "Wintergreen"), 
                     values = c("blue", "red", "black", "forestgreen","violet","limegreen", "purple", "orange", "maroon1", "turquoise3", "thistle4")) +
  facet_grid(trophicstate ~ Variable, scales = "free",labeller = profile_labeller, space = "free_y") +  
  theme(strip.background = element_blank(),
        strip.text = element_text(face = "bold", colour = "black"),
        axis.title.y = element_text(face="bold"),  #Set the y-axis title
        legend.title = element_text(size = 8, face="bold"),  #Set the legend title 
        legend.text = element_text(size = 8),  #Set the legend text
        legend.position="right"); #c(0.225, 0.22));
#dev.off()
```








```{r session info, eval = TRUE, echo = FALSE}
# Provide the Session Information for reproducibility
devtools::session_info()
```